---
output:
  html_document:
    df_print: paged
    fig.align: center
    self_contained: yes 
    fig.height: 4
    fig.width: 8
    theme: united
    toc: yes
    toc_depth: 4
    toc_float: yes
    number_sections: yes
    code_folding: hide
title: "Auswertung TC Daten"
author: "David Lilek"
date: "`r format(Sys.time(), '%d %B %Y, %X')`"
editor_options: 
  markdown: 
    wrap: 72
---


# Load Libraries & Plot function

```{r message=FALSE, warning=FALSE}
library(ggplot2)
library(tidyverse)
library(dplyr)
library(UpSetR)
library(pheatmap)
library(ggvenn)
library(knitr)
library(reshape2)
library(eulerr)
#library(venneuler)
#library(VennDiagram) 
```

```{r}
euler_venn_2comp <- function(lst, label_venn, color_venn, cex_venn, grid, gList, h_just, v_just) {
  ######################################
  #################plot 2 venn diagrams
  ######################################
  p<-plot(euler(lst),
          input="union",
          shape="ellipse",
          quantities=list(type = c("percent", 'counts'),
                          cex = cex_venn), #schriftgröße für A, B bzw intersection
          key=TRUE,
          labels=FALSE,
          lty = c(1,1),
          legend = list(labels = label_venn,
                        side = "bottom",
                        nrow = 1, ncol = 3),
          edges=TRUE,
          fills = color_venn, #color für A,B bzw intersection 
          alpha=0.8)
  # https://stackoverflow.com/questions/75177293/how-to-change-the-label-position-when-plotting-venn-diagram-from-eulerr-package
  tags <- p$children[[2]]$children[[1]]$children$tags$children
  tags <- do.call(grid::gList, lapply(tags, function(x) {
    x$children[[2]]$label <- sub(" \\(", "\n(", x$children[[2]]$label)
    x$children[[2]]$just <- NULL
    x$children[[2]]$hjust <- h_just
    x$children[[2]]$vjust <- v_just
    x}))
  
  p$children[[2]]$children[[1]]$children$tags$children <- tags
  print(p)
}


euler_venn_3comp <- function(lst, cex_venn, label_venn, color_venn, grid, gList, h_just, v_just) {
  ##################################
  ###### venn diagram 3 comparisons
  ##################################
  p<-plot(euler(lst),
          input="union",
          shape="ellipse",
          quantities=list(type = c("percent", 'counts'),
                          cex = cex_venn), #schriftgröße für A, B, C bzw intersection
          key=TRUE,
          labels=FALSE,
          lty = c(1,1,1),
          legend = list(labels = label_venn,
                        side = "bottom",
                        nrow = 1, ncol = 3),
          edges=TRUE,
          fills = color_venn, #color für A,B bzw intersection fills = grey[c(10,18,25)]
          alpha=0.8)
  # https://stackoverflow.com/questions/75177293/how-to-change-the-label-position-when-plotting-venn-diagram-from-eulerr-packages
  tags <- p$children[[2]]$children[[1]]$children$tags$children
  tags <- do.call(grid::gList, lapply(tags, function(x) {
    x$children[[2]]$label <- sub(" \\(", "\n(", x$children[[2]]$label)
    x$children[[2]]$just <- NULL
    x$children[[2]]$hjust <- h_just
    x$children[[2]]$vjust <- v_just
    x}))
  
  p$children[[2]]$children[[1]]$children$tags$children <- tags
  print(p)
}

```


# Vergleich Messreplikate Standardabweichung

* Taking a closer look the standard deviation of measurement replicates it is obvious that the standard deviation depends stronly on the number of identified proteins. Below 200 proteins a rsd of 5.24% was calculated. Above 200 proteins a significantly lower rsd of 1.57% was observed (p-value = 0.001608). 
* die Messungen wurden in Triplikaten und Duplikaten durchgeführt
* für jedes Triplikat/Duplikat wurde sd und mean berechnet und geplottet
* es zeigt sich, dass die Messreproduzierbarkeit von der Anzahl der Proteine abhängt
* <200 Proteine: median relative sd = 5.235318
* \>200 proteine: median realtive sd = 1.568858

```{r}
data <- read.csv("Extraktionen_gesamt_V2_DL.csv", sep=";")
sample_group <- c(seq(1,18,by=3),seq(19,162,by=2))
sd_samples <- c()
mean_samples <- c()
for (i in 1:length(sample_group)){
  if (i<78){
  tmp <- sd(data$Unique.no.MBR.2.peptides[sample_group[i]:(sample_group[i+1]-1)])
  sd_samples <- c(sd_samples, tmp)
  tmp <- mean(data$Unique.no.MBR.2.peptides[sample_group[i]:(sample_group[i+1]-1)])
  mean_samples <- c(mean_samples,tmp)
  } else {
    tmp <- sd(data$Unique.no.MBR.2.peptides[sample_group[i]:length(data$Unique.no.MBR.2.peptides)])
    sd_samples <- c(sd_samples, tmp)
    tmp <- mean(data$Unique.no.MBR.2.peptides[sample_group[i]:length(data$Unique.no.MBR.2.peptides)])
    mean_samples <- c(mean_samples,tmp)
  }
}

plot(sd_samples[-31]/mean_samples[-31]*100,main="relative standard deviation",ylab="rel sd [%]",xlab="sample number")
plot(mean_samples[-31],sd_samples[-31]/mean_samples[-31]*100,ylab="rel. sd [%]",xlab="number of identified proteins")

rel_sd <- sd_samples/mean_samples*100
median(rel_sd[which(mean_samples<200)])
median(rel_sd[which(mean_samples>200)])
#t test rel sd <200 with rel sd >200
t.test(rel_sd[which(mean_samples<200)],rel_sd[which(mean_samples>200)])
```


# Comparison E3_D1 - noTreatment, Heatshock, ProteaseInhibitor

The results show that the significant highest number of protein identifications (median=586) could be achieved using protease inhibitor. The significant lowest number of proteins was determined using heatshock (median=315). No treatment resulted in 491 identified proteins. (p value ANOVA 1.43e-09; p value Tukey HSD test below 0.0003). All measurement were performed in using 6 technical replicates.


ACHTUNG: Die Anzahl der oben genannten Proteine beziehen sich auf die Bildung des einfachen Mittelwertes. Hier wenn man merged und die "wirkliche" Anzahl an Proteinen gibt.

```{r message=FALSE, warning=FALSE}
#E3_D1
D1_raw <- readRDS(file = "results_v002/results_run1_mqpar_extracts_2gether_E3_D1_LFQ_combined_txt_proteinGroups.txt_Unique.RDS")
D1_raw <- D1_raw[[1]]

D1 <- D1_raw[,-ncol(D1_raw)]
D1 <- D1*1

noT <- apply(D1[,c(1:6)], 1, any)
HS <- apply(D1[,c(7:12)],1,any)
PI <- apply(D1[,c(13:18)],1,any)
print("no Treatment")
sum(noT*1)
print("Heatshock")
sum(HS*1)
print("Protease Inhibitor")
sum(PI*1)
```

```{r}
data <- read.csv("Extraktionen_gesamt_V2_DL.csv", sep=";")
data_sub <- as.data.frame(data$Unique.no.MBR.2.peptides)
data_sub <- as.data.frame(data_sub[c(1:18),])
data_sub$group <- c(rep("noTreatment",6),
                   rep("Heatshock",6),
                   rep("ProteaseInhibitor",6))
colnames(data_sub) <- c("number","group")
print("Median no Treatment")
median(subset(data_sub,group=="noTreatment")[,1])
print("Median Heatshock")
median(subset(data_sub,group=="Heatshock")[,1])
print("Median ProteaseInhibitor")
median(subset(data_sub,group=="ProteaseInhibitor")[,1])
ggplot(data_sub, aes(x=group, y=number)) + 
  geom_boxplot()
anova <- aov(number ~ group, data=data_sub)
summary(anova)
Tukey <- TukeyHSD(anova)
plot(Tukey)
Tukey
```

# E4 - D1 - B - AF - TF

```{r}

############## B
data <- readRDS(file = "results_v002/results_run1_mqpar_extracts_2gether_E4_D1_B_LFQ_combined_txt_proteinGroups.txt_Unique.RDS")
data <- data[[1]]
data_sub <- dplyr::select(data, contains("Pool_G"))
#get rid of FALSE values
B <- data_sub*1
B[B==0] <- NA
ind <- apply(B, 1, function(x) all(is.na(x)))


############AF
data <- readRDS(file = "results_v002/results_run1_mqpar_extracts_2gether_E4_D1_AF_TF_LFQ_combined_txt_proteinGroups.txt_Unique.RDS")
data <- data[[1]]
data_sub <- dplyr::select(data, contains("Pool_AF"))
#get rid of FALSE values
AF <- data_sub*1
AF[AF==0] <- NA
ind <- apply(AF, 1, function(x) all(is.na(x)))

############TF
data <- readRDS(file = "results_v002/results_run1_mqpar_extracts_2gether_E4_D1_AF_TF_LFQ_combined_txt_proteinGroups.txt_Unique.RDS")
data <- data[[1]]
data_sub <- dplyr::select(data, contains("Pool_TF"))
#get rid of FALSE values
TF <- data_sub*1
TF[TF==0] <- NA
ind <- apply(TF, 1, function(x) all(is.na(x)))


## compare AF TF B poster
AF_TF_raw <- readRDS(file = "results_v002/results_run1_mqpar_extracts_2gether_E4_D1_AF_TF_LFQ_combined_txt_proteinGroups.txt_Unique.RDS")
AF_TF_raw <- AF_TF_raw[[1]]
AF <- dplyr::select(AF_TF_raw, contains("AF"))
AF <- apply(AF, 1, any)
fasta <- c()
for (i in 1:length(AF_TF_raw$FASTA)){
  tmp <- strsplit(AF_TF_raw$FASTA[i],split='\\|')[[1]][2]
  fasta <- c(fasta,tmp)
}
AF <- as.data.frame(cbind(AF, fasta))
TF <- dplyr::select(AF_TF_raw, contains("TF"))
TF <- apply(TF, 1, any)
TF <- as.data.frame(cbind(TF, fasta))
B_raw <- readRDS(file = "results_v002/results_run1_mqpar_extracts_2gether_E4_D1_B_LFQ_combined_txt_proteinGroups.txt_Unique.RDS")
B_raw <- B_raw[[1]]
B <- dplyr::select(B_raw, contains("Pool_G"))
B <- apply(B, 1, any)
fasta <- c()
for (i in 1:length(B_raw$FASTA)){
  tmp <- strsplit(B_raw$FASTA[i],split='\\|')[[1]][2]
  fasta <- c(fasta,tmp)
}
B <- as.data.frame(cbind(B, fasta))

AF_TF_B_comparison  <- merge(B, AF,
                           all = TRUE,
                           by = "fasta")
AF_TF_B_comparison <- merge(AF_TF_B_comparison, TF,
                            all = TRUE,
                            by = "fasta")

AF_TF_B_comparison$B <- as.logical(AF_TF_B_comparison$B)
AF_TF_B_comparison$AF <- as.logical(AF_TF_B_comparison$AF)
AF_TF_B_comparison$TF <- as.logical(AF_TF_B_comparison$TF)
AF_TF_B_comparison <- AF_TF_B_comparison[,-1]*1
AF_TF_B_comparison[AF_TF_B_comparison==0] <- NA


# boxplot
boxplot(sum(AF_TF_B_comparison$B,na.rm = TRUE),
        sum(AF_TF_B_comparison$AF,na.rm = TRUE),
        sum(AF_TF_B_comparison$TF,na.rm = TRUE),
        names = c("E4_D1_B","E4_D1_AF","E4_D1_TF"))

proteins <- as.data.frame(which(!is.na(AF_TF_B_comparison), arr.ind=TRUE))

B <- proteins %>% dplyr::filter(col == 1)
AF <- proteins %>% dplyr::filter(col == 2)
TF <- proteins %>% dplyr::filter(col == 3)


# combine to list
lst <- list("E4_D1_B"=as.double(B[,1]),
            "E4_D1_AF"=as.double(AF[,1]),
            "E4_D1_TF"=as.double(TF[,1]))

upset(fromList(lst), 
      order.by = "freq",
      nsets = 10,
      nintersects = 15,
      keep.order = TRUE,
      sets=c("E4_D1_B", "E4_D1_AF", "E4_D1_TF"),
      mainbar.y.label = "Number of shared proteins",
      sets.x.label = "Number of proteins per fraction",
      set_size.show = TRUE,
      set_size.angles = 0,
      set_size.scale_max = 1800
)


#get colors
#total protein extract
total <- rgb(255,93,93, maxColorValue = 255)
#fractions high MW SEC
highMW <- rgb(169,208,142,maxColorValue = 255)
#fractions low MW SEC
lowMW <- rgb(255,217,102, maxColorValue = 255)
#in-gel digest
IG <- rgb(180,198,231,maxColorValue = 255)

ggvenn(lst,
       fill_color = c(IG,lowMW,highMW),
       stroke_size = 0.5, set_name_size = 5, text_size = 4.25, digits = 0, fill_alpha = 0.9, label_sep = )
#ggsave("./pics/unique_AF_TF_B.png")
euler_venn_3comp(lst, label_venn = c("E4_D1_B","E4_D1_AF","E4_D1_TF"), color_venn = c(IG,lowMW,highMW), cex_venn = c(1,1,1), h_just = 0.5, v_just = 0.5)

print(c("E4_D1_B","E4_D1_AF","E4_D1_TF"))
sum(AF_TF_B_comparison$B,na.rm = TRUE)
sum(AF_TF_B_comparison$AF,na.rm = TRUE)
sum(AF_TF_B_comparison$TF,na.rm = TRUE)
        

#compare B to pool from 0609 
B_raw <- readRDS(file = "results_v002/results_run1_mqpar_extracts_2gether_E4_D1_B_LFQ_combined_txt_proteinGroups.txt_Unique.RDS")
B_raw <- B_raw[[1]]
B <- dplyr::select(B_raw, contains("Pool_G"))
B <- apply(B, 1, any)
fasta <- c()
for (i in 1:length(B_raw$FASTA)){
  tmp <- strsplit(B_raw$FASTA[i],split='\\|')[[1]][2]
  fasta <- c(fasta,tmp)
}
B <- as.data.frame(cbind(B, fasta))
aj_raw <- readRDS(file = "results_v002/results_run1_mqpar_extracts_2gether_E4_D1_a-j_LFQ_combined_txt_proteinGroups.txt_Unique.RDS")
aj_raw <- aj_raw[[1]]
aj <- dplyr::select(aj_raw, contains("Pool"))
aj <- apply(aj, 1, any)
fasta <- c()
for (i in 1:length(aj_raw$FASTA)){
  tmp <- strsplit(aj_raw$FASTA[i],split='\\|')[[1]][2]
  fasta <- c(fasta,tmp)
}
aj <- as.data.frame(cbind(aj, fasta))
B_aj_comparison  <- merge(B, aj,
                             all = TRUE,
                             by = "fasta")

B_aj_comparison$B <- as.logical(B_aj_comparison$B)
B_aj_comparison$aj <- as.logical(B_aj_comparison$aj)
B_aj_comparison <- B_aj_comparison[,-1]*1
B_aj_comparison[B_aj_comparison==0] <- NA
boxplot(sum(B_aj_comparison$B,na.rm = TRUE),
        sum(B_aj_comparison$aj,na.rm = TRUE),
        names = c("B","E4_D1_pool"))

proteins <- as.data.frame(which(!is.na(B_aj_comparison), arr.ind=TRUE))

B_inter <- proteins %>% dplyr::filter(col == 1)
aj_inter <- proteins %>% dplyr::filter(col == 2)

# combine to list
lst <- list("E4_D1_B_fractions"=as.double(B_inter[,1]),
            "E4_D1"=as.double(aj_inter[,1]))

upset(fromList(lst), 
      order.by = "freq",
      nsets = 10,
      nintersects = 15,
      keep.order = TRUE,
      sets=c("E4_D1_B_fractions", "E4_D1"),
      mainbar.y.label = "Number of shared proteins",
      sets.x.label = "Number of proteins per fraction",
      set_size.show = TRUE,
      set_size.angles = 0,
      set_size.scale_max = 1800
)

ggvenn(lst,
       fill_color = c(IG,total),
       stroke_size = 0.5, set_name_size = 5, text_size = 4, digits = 0, fill_alpha = 0.9)
#ggsave("./pics/unique_B_D1pool.png")

euler_venn_2comp(lst, label_venn = c("E4_D1_B_fractions","E4_D1"), color_venn = c(IG,total), cex_venn = c(1,1), h_just = 0.5, v_just = 0.5)

print(c("B","E4_D1_pool"))
sum(B_aj_comparison$B,na.rm = TRUE)
sum(B_aj_comparison$aj,na.rm = TRUE)
       

```


## compare E4_D1_Pool_G_B1-B9 and E4_D1_Pool_20_B1-B9

* um Unterschiede zwischen B1-B9 in Pool_G und Pool_20 zu sehen
* habe mir die Proteine auch angeschaut die unterschiedlich sind. 31 davon sind uncharacterized proteins
* teilweise sind sich die proteine schon ähnlich zumindest von der accesion number

```{r}
#E4_D1_Pool_G_B1-B9
data <- readRDS(file = "results_v002/results_run1_mqpar_extracts_2gether_E4_D1_B_LFQ_combined_txt_proteinGroups.txt_Unique.RDS")
data <- data[[1]]
data_sub <- dplyr::select(data, contains("Pool_G"))
B_D1<-apply(data_sub, 1, any)
fasta <- c()
for (i in 1:length(data$FASTA)){
  tmp <- strsplit(data$FASTA[i],split='\\|')[[1]][2]
  fasta <- c(fasta,tmp)
}
B_D1 <- as.data.frame(cbind(B_D1, fasta))
#B_D1 <- as.data.frame(cbind(B_D1, data$FASTA))

#E4_D1_Pool_20_B1-B9

D1_raw <- readRDS(file = "results_v002/results_run1_mqpar_extracts_2gether_E4_D1_B_LFQ_combined_txt_proteinGroups.txt_Unique.RDS")
D1_raw <- D1_raw[[1]]
D1 <- dplyr::select(D1_raw, contains("20"))
D1 <- apply(D1, 1, any)
#D1 <- as.data.frame(cbind(D1, D1_raw$FASTA))
fasta <- c()
for (i in 1:length(data$FASTA)){
  tmp <- strsplit(data$FASTA[i],split='\\|')[[1]][2]
  fasta <- c(fasta,tmp)
}
D1 <- as.data.frame(cbind(D1, fasta))

G_pool_comparison  <- merge(B_D1, D1,
                           all = TRUE,
                           by = "fasta")

G_pool_comparison$B_D1 <- as.logical(G_pool_comparison$B_D1)
G_pool_comparison$D1 <- as.logical(G_pool_comparison$D1)
G_pool_comparison <- G_pool_comparison[,-1]*1
G_pool_comparison[G_pool_comparison==0] <- NA

# boxplot
boxplot(sum(G_pool_comparison$B_D1,na.rm = TRUE),
        sum(G_pool_comparison$D1,na.rm = TRUE),
        names = c("E4_D1_G","E4_D1_20"))

proteins <- as.data.frame(which(!is.na(G_pool_comparison), arr.ind=TRUE))

B_D1_inter <- proteins %>% dplyr::filter(col == 1)
D1_inter <- proteins %>% dplyr::filter(col == 2)

# combine to list
lst <- list("E4_D1_G"=as.double(B_D1_inter[,1]),
            "E4_D1_20"=as.double(D1_inter[,1]))

upset(fromList(lst), 
      order.by = "freq",
      nsets = 10,
      nintersects = 15,
      keep.order = TRUE,
      sets=c("E4_D1_G", "E4_D1_20"),
      mainbar.y.label = "Number of shared proteins",
      sets.x.label = "Number of proteins per fraction",
      set_size.show = TRUE,
      set_size.angles = 0,
      set_size.scale_max = 1000
)

#venn digarams
#get colors
D1_col <- rgb(255,93,93, maxColorValue = 255)
D2_col <- rgb(169,208,142,maxColorValue = 255)
#E4_D1_pool_col <- rgb(255,217,102, maxColorValue = 255)

ggvenn(lst,
       fill_color = c(D1_col,D2_col),
       stroke_size = 0.5, set_name_size = 4, text_size = 4, digits = 0, fill_alpha = 0.9)
euler_venn_2comp(lst, label_venn = c("E4_D1_G","E4_D1_20"), color_venn = c(D1_col,D2_col), cex_venn = c(1,1), h_just = 0.5, v_just = 0.5)

print(c("E4_D1_G","E4_D1_20"))
sum(G_pool_comparison$B_D1,na.rm = TRUE)
sum(G_pool_comparison$D1,na.rm = TRUE)
```

# Compare E3e E3f with E4_D1_a-j with E5_a_b

-   E3 e bzw E3 f sollte dasselbe sein wie das hier
    `file:///X:/wissenschaftliche%20Ver%C3%B6ffentlichungen/Publikationen/2022/Proteomics_JRS_DL/data/results_v002/results_run1_mqpar_extracts_2gether_E4_D1_a-j_LFQ_combined_txt_proteinGroups.txt_post-processing.html#52_Summary_Unique`
    und auch E5_D1 a und b

```{r}
aj_raw <- readRDS(file = "results_v002/results_run1_mqpar_extracts_2gether_E4_D1_a-j_LFQ_combined_txt_proteinGroups.txt_Unique.RDS")
aj_raw <- aj_raw[[1]]
aj <- aj_raw[,-ncol(aj_raw)]
FASTA <- c()
for (i in 1:length(aj_raw$FASTA)){
  tmp <- strsplit(aj_raw$FASTA[i],split='\\|')[[1]][2]
  FASTA<- c(FASTA,tmp)
}
aj <- cbind(aj,FASTA)

E3_D1_raw <- readRDS(file = "results_v002/results_run1_mqpar_extracts_2gether_E3_D1_LFQ_combined_txt_proteinGroups.txt_Unique.RDS")
E3_D1_raw <- E3_D1_raw[[1]]
E3_D1 <- E3_D1_raw[,c(13:18)]
FASTA <- c()
for (i in 1:length(E3_D1_raw$FASTA)){
  tmp <- strsplit(E3_D1_raw$FASTA[i],split='\\|')[[1]][2]
  FASTA<- c(FASTA,tmp)
}
E3_D1 <- cbind(E3_D1,FASTA)

E5_D1_raw <- readRDS(file="./27_20230201_FH/results/D5_extracts_Unique.RDS")
E5_D1_raw <- E5_D1_raw[[1]]
E5_D1 <- dplyr::select(E5_D1_raw, contains(".1"))
colnames(E5_D1) <- c(paste("Unique.peptides.","E5_D",substring(colnames(E5_D1),17),sep=""))
FASTA <- c()
for (i in 1:length(E5_D1_raw$FASTA)){
  tmp <- strsplit(E5_D1_raw$FASTA[i],split='\\|')[[1]][2]
  FASTA<- c(FASTA,tmp)
}
E5_D1 <- cbind(E5_D1,FASTA)


comp  <- merge(aj, E3_D1,
               all = TRUE,
               by = "FASTA")
comp <- merge(comp, E5_D1,
              all=TRUE,
              by="FASTA")


paletteLength <- 2
myColor <- colorRampPalette(c("navy", "white",
                              "firebrick3"))(paletteLength)
data4pheatmap <- comp[,-1]
data4pheatmap <- data4pheatmap*1
data4pheatmap[data4pheatmap == 0] <- NA
# remove NA rows
ind <- apply(data4pheatmap, 1, function(x) all(is.na(x)))
data4pheatmap_clear <- data4pheatmap[!ind,]
colnames(data4pheatmap_clear) <- colnames(comp[-1])
data4pheatmap_clear[is.na(data4pheatmap_clear)] <- 0
pheatmap(data4pheatmap_clear,
         legend_breaks = c(0,1),
         color = myColor,
         treeheight_row = 10,
         angle_col ="45",
         treeheight_col = 10,
         legend = TRUE,
         labels_row = rep("",nrow(data4pheatmap)),
         labels_col = substring(colnames(data4pheatmap_clear),17))
#plot results
results_2gether <- sapply(comp[,-1],function(x)sum(x, na.rm = TRUE))
print(kable(results_2gether))
ggplot(melt(results_2gether),aes(x = rownames(melt(results_2gether)), y = value)) +
  geom_boxplot() + 
  xlab("Sample name") +
  theme(axis.text.x = element_text(angle = 90)) +
  labs(title = "Unique")
```


# E4-D2

```{r}
#############################
# D2
###############################
data <- readRDS(file = "results_v002/results_run1_mqpar_extracts_2gether_D2_LFQ_combined_txt_proteinGroups.txt_Unique.RDS")
data <- data[[1]]
data_sub <- dplyr::select(data, contains("Pool_20"))
B_D2 <- data_sub*1
B_D2[B_D2==0] <- NA


### analysis B
proteins <- as.data.frame(which(!is.na(B_D2), arr.ind=TRUE))

B1 <- proteins %>% dplyr::filter(col == 1 ) # | col == 2 -> means or col2
B10 <- proteins %>% dplyr::filter(col == 2)
B2 <- proteins %>% dplyr::filter(col == 3)
B3 <- proteins %>% dplyr::filter(col == 4)
B4 <- proteins %>% dplyr::filter(col == 5)
B5 <- proteins %>% dplyr::filter(col == 6)
B6 <- proteins %>% dplyr::filter(col == 7)
B7 <- proteins %>% dplyr::filter(col == 8)
B8 <- proteins %>% dplyr::filter(col == 9)
B9 <- proteins %>% dplyr::filter(col == 10)
# combine to list
lst <- list(B1 = as.double(B1[,1]),
            B2 = as.double(B2[,1]),
            B3 = as.double(B3[,1]),
            B4 = as.double(B4[,1]),
            B5 = as.double(B5[,1]),
            B6 = as.double(B6[,1]),
            B7 = as.double(B7[,1]),
            B8 = as.double(B8[,1]),
            B9 = as.double(B9[,1]),
            B10 = as.double(B10[,1]))            

upset(fromList(lst), 
      order.by = "freq",
      nsets = 9,
      nintersects = 15,
      sets = c("B1","B2","B3","B4","B5","B6","B7","B8","B9","B10"),
      keep.order = TRUE,
      mainbar.y.label = "Number of shared proteins",
      sets.x.label = "Number of proteins per fraction",
      set_size.show = TRUE,
      set_size.angles = 0,
      set_size.scale_max = 1000
)


data_D2_pool <- dplyr::select(data, contains("Gesamt"))
data_D2_pool <- apply(data_D2_pool, 1, any)
data_D2_fractions <- dplyr::select(data, contains("20"))
data_D2_fractions <- apply(data_D2_fractions, 1, any)
data_D2 <- as.data.frame(cbind(data_D2_pool,data_D2_fractions))
data_D2 <- data_D2*1
data_D2[data_D2==0] <- NA
# boxplot
boxplot(sum(data_D2$data_D2_pool,na.rm = TRUE),
        sum(data_D2$data_D2_fractions,na.rm = TRUE),
        names = c("D2 pool","D2 fractions"))


proteins <- as.data.frame(which(!is.na(data_D2), arr.ind=TRUE))

D2_pool_inter <- proteins %>% dplyr::filter(col == 1)
D2_fractions_inter <- proteins %>% dplyr::filter(col == 2)

# combine to list
lst <- list(D2_pool=as.double(D2_pool_inter[,1]),
            D2_fractions=as.double(D2_fractions_inter[,1]))

upset(fromList(lst), 
      order.by = "freq",
      nsets = 10,
      nintersects = 15,
      keep.order = TRUE,
      sets=c("D2_pool", "D2_fractions"),
      mainbar.y.label = "Number of shared proteins",
      sets.x.label = "Number of proteins per fraction",
      set_size.show = TRUE,
      set_size.angles = 0,
      set_size.scale_max = 1000
)

#venn digarams
#get colors
D2_pool_col <- rgb(255,93,93, maxColorValue = 255)
D2_ingel_col <- rgb(169,208,142,maxColorValue = 255)
#E4_D1_pool_col <- rgb(255,217,102, maxColorValue = 255)

ggvenn(lst,
       fill_color = c(D2_pool_col,D2_ingel_col),
       stroke_size = 0.5, set_name_size = 5, text_size = 4, digits = 0, fill_alpha = 0.9)

euler_venn_2comp(lst, label_venn = c("D2_pool","D2_fractions"), color_venn = c(D2_pool_col,D2_ingel_col), cex_venn = c(1,1), h_just = 0.5, v_just = 0.5)

```

# E4 - Compare D1-D2

## fractions

### E4_D1_Pool_G_B1-B9 with E4_D2_Pool_20_B1-B10

```{r}
#################### D1 fractions
data <- readRDS(file = "results_v002/results_run1_mqpar_extracts_2gether_E4_D1_B_LFQ_combined_txt_proteinGroups.txt_Unique.RDS")
data <- data[[1]]
data_sub <- dplyr::select(data, contains("Pool_G"))
B_D1<-apply(data_sub, 1, any)
fasta <- c()
for (i in 1:length(data$FASTA)){
  tmp <- strsplit(data$FASTA[i],split='\\|')[[1]][2]
  fasta <- c(fasta,tmp)
}
B_D1 <- as.data.frame(cbind(B_D1, fasta))
#B_D1 <- as.data.frame(cbind(B_D1, data$FASTA))


#################### D2 fractions
data <- readRDS(file = "results_v002/results_run1_mqpar_extracts_2gether_D2_LFQ_combined_txt_proteinGroups.txt_Unique.RDS")
data <- data[[1]]
data_sub <- dplyr::select(data, contains("Pool_20"))
B_D2<-apply(data_sub, 1, any)
fasta <- c()
for (i in 1:length(data$FASTA)){
  tmp <- strsplit(data$FASTA[i],split='\\|')[[1]][2]
  fasta <- c(fasta,tmp)
}
B_D2 <- as.data.frame(cbind(B_D2, fasta))
#B_D2 <- as.data.frame(cbind(B_D2, data$FASTA))
#D1_D2_comparison  <- merge(B_D1, B_D2,
#                           all = TRUE,
#                           by = "V2")
D1_D2_comparison  <- merge(B_D1, B_D2,
                           all = TRUE,
                           by = "fasta")

D1_D2_comparison$B_D1 <- as.logical(D1_D2_comparison$B_D1)
D1_D2_comparison$B_D2 <- as.logical(D1_D2_comparison$B_D2)
D1_D2_comparison <- D1_D2_comparison[,-1]*1
D1_D2_comparison[D1_D2_comparison==0] <- NA

# boxplot
boxplot(sum(D1_D2_comparison$B_D1,na.rm = TRUE),
        sum(D1_D2_comparison$B_D2,na.rm = TRUE),
        names = c("E4_D1_fractions","E4_D2_fractions"),
        main="Comparison D1/D2 fractions")
#upset r & ggvenn
proteins <- as.data.frame(which(!is.na(D1_D2_comparison), arr.ind=TRUE))

D1_inter <- proteins %>% dplyr::filter(col == 1)
D2_inter <- proteins %>% dplyr::filter(col == 2)

# combine to list
lst <- list("E4_D1_fraction"=as.double(D1_inter[,1]),
            "E4_D2_fraction"=as.double(D2_inter[,1]))

upset(fromList(lst), 
      order.by = "freq",
      nsets = 10,
      nintersects = 15,
      keep.order = TRUE,
      sets=c("E4_D1_fraction", "E4_D2_fraction"),
      mainbar.y.label = "Number of shared proteins",
      sets.x.label = "Number of proteins per fraction",
      set_size.show = TRUE,
      set_size.angles = 0,
      set_size.scale_max = 1000
)

#venn digarams
#get colors
D1_col <- rgb(255,93,93, maxColorValue = 255)
D2_col <- rgb(169,208,142,maxColorValue = 255)
#E4_D1_pool_col <- rgb(255,217,102, maxColorValue = 255)

ggvenn(lst,
       fill_color = c(D1_col,D2_col),
       stroke_size = 0.5, set_name_size = 4, text_size = 4, digits = 0, fill_alpha = 0.9)

euler_venn_2comp(lst, label_venn = c("E4_D1_fraction","E4_D2_fraction"), color_venn = c(D1_col,D2_col), cex_venn = c(1,1), h_just = 0.5, v_just = 0.5)

print(c("E4_D1_fractions","E4_D2_fractions"))
sum(D1_D2_comparison$B_D1,na.rm = TRUE)
sum(D1_D2_comparison$B_D2,na.rm = TRUE)
        
```

### E4_D1_Pool_20_B1-B9 with E4_D2_Pool_20_B1-B10

```{r}
# D1-D2 comparison

D1_raw <- readRDS(file = "results_v002/results_run1_mqpar_extracts_2gether_E4_D1_B_LFQ_combined_txt_proteinGroups.txt_Unique.RDS")
D1_raw <- D1_raw[[1]]
D1 <- dplyr::select(D1_raw, contains("20"))
D1 <- apply(D1, 1, any)
#D1 <- as.data.frame(cbind(D1, D1_raw$FASTA))
fasta <- c()
for (i in 1:length(data$FASTA)){
  tmp <- strsplit(data$FASTA[i],split='\\|')[[1]][2]
  fasta <- c(fasta,tmp)
}
D1 <- as.data.frame(cbind(D1, fasta))
D2_raw <- readRDS(file = "results_v002/results_run1_mqpar_extracts_2gether_D2_LFQ_combined_txt_proteinGroups.txt_Unique.RDS")
D2_raw <- D2_raw[[1]]
D2 <- dplyr::select(D2_raw, contains("Pool_20"))
D2 <- apply(D2, 1, any)
fasta <- c()
for (i in 1:length(data$FASTA)){
  tmp <- strsplit(data$FASTA[i],split='\\|')[[1]][2]
  fasta <- c(fasta,tmp)
}
#D2 <- as.data.frame(cbind(D2, D2_raw$FASTA))
D2 <- as.data.frame(cbind(D2, fasta))
D1_D2_comparison  <- merge(D1, D2,
                           all = TRUE,
                           by = "fasta")

D1_D2_comparison$D1 <- as.logical(D1_D2_comparison$D1)
D1_D2_comparison$D2 <- as.logical(D1_D2_comparison$D2)
D1_D2_comparison <- D1_D2_comparison[,-1]*1
D1_D2_comparison[D1_D2_comparison==0] <- NA

# boxplot
boxplot(sum(D1_D2_comparison$D1,na.rm = TRUE),
        sum(D1_D2_comparison$D2,na.rm = TRUE),
        names = c("E4_D1_fraction","E4_D2_fraction"))

proteins <- as.data.frame(which(!is.na(D1_D2_comparison), arr.ind=TRUE))

D1_inter <- proteins %>% dplyr::filter(col == 1)
D2_inter <- proteins %>% dplyr::filter(col == 2)

# combine to list
lst <- list("E4_D1_fraction"=as.double(D1_inter[,1]),
            "E4_D2_fraction"=as.double(D2_inter[,1]))

upset(fromList(lst), 
      order.by = "freq",
      nsets = 10,
      nintersects = 15,
      keep.order = TRUE,
      sets=c("E4_D1_fraction", "E4_D2_fraction"),
      mainbar.y.label = "Number of shared proteins",
      sets.x.label = "Number of proteins per fraction",
      set_size.show = TRUE,
      set_size.angles = 0,
      set_size.scale_max = 1000
)

#venn digarams
#get colors
D1_col <- rgb(255,93,93, maxColorValue = 255)
D2_col <- rgb(169,208,142,maxColorValue = 255)
#E4_D1_pool_col <- rgb(255,217,102, maxColorValue = 255)

ggvenn(lst,
       fill_color = c(D1_col,D2_col),
       stroke_size = 0.5, set_name_size = 4, text_size = 4, digits = 0, fill_alpha = 0.9)
euler_venn_2comp(lst, label_venn = c("E4_D1_fraction","E4_D2_fraction"), color_venn = c(D1_col,D2_col), cex_venn = c(1,1), h_just = 0.5, v_just = 0.5)

print(c("E4_D1_fraction","E4_D2_fraction"))
sum(D1_D2_comparison$D1,na.rm = TRUE)
sum(D1_D2_comparison$D2,na.rm = TRUE)

```

## pool

```{r}
# D1-D2 comparison

D1_raw <- readRDS(file = "results_v002/results_run1_mqpar_extracts_2gether_E4_D1_a-j_LFQ_combined_txt_proteinGroups.txt_Unique.RDS")
D1_raw <- D1_raw[[1]]
D1 <- dplyr::select(D1_raw, contains("Pool"))
D1 <- apply(D1, 1, any)
#D1 <- as.data.frame(cbind(D1, D1_raw$FASTA))
fasta <- c()
for (i in 1:length(D1_raw$FASTA)){
  tmp <- strsplit(D1_raw$FASTA[i],split='\\|')[[1]][2]
  fasta <- c(fasta,tmp)
}
D1 <- as.data.frame(cbind(D1, fasta))
D2_raw <- readRDS(file = "results_v002/results_run1_mqpar_extracts_2gether_D2_LFQ_combined_txt_proteinGroups.txt_Unique.RDS")
D2_raw <- D2_raw[[1]]
D2 <- dplyr::select(D2_raw, contains("Gesamt"))
D2 <- apply(D2, 1, any)
fasta <- c()
for (i in 1:length(D2_raw$FASTA)){
  tmp <- strsplit(D2_raw$FASTA[i],split='\\|')[[1]][2]
  fasta <- c(fasta,tmp)
}
#D2 <- as.data.frame(cbind(D2, D2_raw$FASTA))
D2 <- as.data.frame(cbind(D2, fasta))
D1_D2_comparison  <- merge(D1, D2,
                           all = TRUE,
                           by = "fasta")

D1_D2_comparison$D1 <- as.logical(D1_D2_comparison$D1)
D1_D2_comparison$D2 <- as.logical(D1_D2_comparison$D2)
D1_D2_comparison <- D1_D2_comparison[,-1]*1
D1_D2_comparison[D1_D2_comparison==0] <- NA

# boxplot
boxplot(sum(D1_D2_comparison$D1,na.rm = TRUE),
        sum(D1_D2_comparison$D2,na.rm = TRUE),
        names = c("E4_D1","E4_D2"))

proteins <- as.data.frame(which(!is.na(D1_D2_comparison), arr.ind=TRUE))

D1_inter <- proteins %>% dplyr::filter(col == 1)
D2_inter <- proteins %>% dplyr::filter(col == 2)

# combine to list
lst <- list("E4_D1"=as.double(D1_inter[,1]),
            "E4_D2"=as.double(D2_inter[,1]))

upset(fromList(lst), 
      order.by = "freq",
      nsets = 10,
      nintersects = 15,
      keep.order = TRUE,
      sets=c("E4_D1", "E4_D2"),
      mainbar.y.label = "Number of shared proteins",
      sets.x.label = "Number of proteins per fraction",
      set_size.show = TRUE,
      set_size.angles = 0,
      set_size.scale_max = 1000
)

#venn digarams
#get colors
D1_col <- rgb(255,93,93, maxColorValue = 255)
D2_col <- rgb(169,208,142,maxColorValue = 255)
#E4_D1_pool_col <- rgb(255,217,102, maxColorValue = 255)

ggvenn(lst,
       fill_color = c(D1_col,D2_col),
       stroke_size = 0.5, set_name_size = 4, text_size = 4, digits = 0, fill_alpha = 0.9)
euler_venn_2comp(lst, label_venn = c("E4_D1","E4_D2"), color_venn = c(D1_col,D2_col), cex_venn = c(1,1), h_just = 0.5, v_just = 0.5)


print(c("E4_D1","E4_D2"))
sum(D1_D2_comparison$D1,na.rm = TRUE)
sum(D1_D2_comparison$D2,na.rm = TRUE)
```

# E5 - venn diagramms of different dimensions

-   `file:///N:/1_A_Bachelor_Master_Intern/2_M_2022/David/Data/27_20230201_FH/post-processing-extracts.html#52_Summary_Unique`
-   venn diagramm 1a,2a,3a,
-   venn diagramm 1b,2b,3b,4b
* aussage: saurer puffer (2a,3b) bringt nichts -> citratpuffer

## a

* E5 a weglassen weil ja auch D4 fehlt

```{r}
data <- readRDS("./27_20230201_FH/results/D5_extracts_Razor.RDS")
data <- data[[1]]
a <- dplyr::select(data, contains("_a"))
D1_a <- a$Razor...unique.peptides.1_a_01+a$Razor...unique.peptides.1_a_02
D2_a <- a$Razor...unique.peptides.2_a_01 + a$Razor...unique.peptides.2_a_02
D3_a <- a$Razor...unique.peptides.3_a_01 + a$Razor...unique.peptides.3_a_02
a <- as.data.frame(cbind(D1_a,D2_a,D3_a))
a[a==0] <- NA
a[a==2] <- 1
boxplot(sum(a$Razor...unique.peptides.1_a_01,na.rm = TRUE),
        sum(a$Razor...unique.peptides.2_a_01,na.rm = TRUE),
        sum(a$Razor...unique.peptides.3_a_01,na.rm = TRUE),
        names = c("E5_D1","E5_D2","E5_D3"),
        main="unique peptides E5_a D1-D3")
#venn diagram
proteins <- as.data.frame(which(!is.na(a), arr.ind=TRUE))
D1_a <- proteins %>% dplyr::filter(col == 1)
D2_a <- proteins %>% dplyr::filter(col == 2)
D3_a <- proteins %>% dplyr::filter(col == 3)

# combine to list
lst <- list(E5_D1_a = as.double(D1_a[,1]),
            E5_D2_a = as.double(D2_a[,1]),
            E5_D3_a = as.double(D3_a[,1]))
#upset r
upset(fromList(lst), 
      order.by = "freq",
      nsets = 9,
      nintersects = 15,
      sets = c("E5_D1_a","E5_D2_a","E5_D3_a"),
      keep.order = TRUE,
      mainbar.y.label = "Number of shared proteins",
      sets.x.label = "Number of proteins per fraction",
      set_size.show = TRUE,
      set_size.angles = 0,
      set_size.scale_max = 1500
)
# venn diagram
ggvenn(lst,
       fill_color = c(IG,highMW,lowMW),
       stroke_size = 0.5, set_name_size = 5, text_size = 4.25, digits = 0, fill_alpha = 0.9, label_sep = )

euler_venn_3comp(lst, label_venn = c("E5_D1_a","E5_D2_a","E5_D3_a"), color_venn = c(IG,highMW,lowMW), cex_venn = c(1,1,1), h_just = 0.5, v_just = 0.5)
```

## b

### D1-D4

```{r}
data <- readRDS("./27_20230201_FH/results/D5_extracts_Razor.RDS")
data <- data[[1]]
b <- dplyr::select(data, contains("_b"))
D1_b <- b$Razor...unique.peptides.1_b_01+ b$Razor...unique.peptides.1_b_02
D2_b <- b$Razor...unique.peptides.2_b_01 + b$Razor...unique.peptides.2_b_02
D3_b <- b$Razor...unique.peptides.3_b_01 + b$Razor...unique.peptides.3_b_02
D4_b <- b$Razor...unique.peptides.3_b_01 + b$Razor...unique.peptides.4_b_02
b <- as.data.frame(cbind(D1_b,D2_b,D3_b,D4_b))
b[b==0] <- NA
b[b==2] <- 1
boxplot(sum(b$Razor...unique.peptides.1_b_01,na.rm = TRUE),
        sum(b$Razor...unique.peptides.2_b_01,na.rm = TRUE),
        sum(b$Razor...unique.peptides.3_b_01,na.rm = TRUE),
        sum(b$Razor...unique.peptides.4_b_01,na.rm = TRUE),
        names = c("E5_D1","E5_D2","E5_D3","E5_D4"),
        main="unique peptides E5_b D1-D4")
#venn diagram
proteins <- as.data.frame(which(!is.na(b), arr.ind=TRUE))
D1_b <- proteins %>% dplyr::filter(col == 1)
D2_b <- proteins %>% dplyr::filter(col == 2)
D3_b <- proteins %>% dplyr::filter(col == 3)
D4_b <- proteins %>% dplyr::filter(col == 4)

# combine to list
lst <- list(E5_D1_b = as.double(D1_b[,1]),
            E5_D2_b = as.double(D2_b[,1]),
            E5_D3_b = as.double(D3_b[,1]),
            E5_D4_b = as.double(D4_b[,1]))
#upset r
upset(fromList(lst), 
      order.by = "freq",
      nsets = 9,
      nintersects = 15,
      sets = c("E5_D1_b","E5_D2_b","E5_D3_b", "E5_D4_b"),
      keep.order = TRUE,
      mainbar.y.label = "Number of shared proteins",
      sets.x.label = "Number of proteins per fraction",
      set_size.show = TRUE,
      set_size.angles = 0,
      set_size.scale_max = 1500
)
# venn diagram
ggvenn(lst,
       stroke_size = 0.5, set_name_size = 5, text_size = 4.25, digits = 0, fill_alpha = 0.9, label_sep = )
```


### D1,D2,D4

```{r}
data <- readRDS("./27_20230201_FH/results/D5_extracts_Razor.RDS")
data <- data[[1]]
b <- dplyr::select(data, contains("_b"))
D1_b <- b$Razor...unique.peptides.1_b_01+ b$Razor...unique.peptides.1_b_02
D2_b <- b$Razor...unique.peptides.2_b_01 + b$Razor...unique.peptides.2_b_02
D4_b <- b$Razor...unique.peptides.3_b_01 + b$Razor...unique.peptides.4_b_02
b <- as.data.frame(cbind(D1_b,D2_b,D4_b))
b[b==0] <- NA
b[b==2] <- 1
boxplot(sum(b$Razor...unique.peptides.1_b_01,na.rm = TRUE),
        sum(b$Razor...unique.peptides.2_b_01,na.rm = TRUE),
        sum(b$Razor...unique.peptides.4_b_01,na.rm = TRUE),
        names = c("E5_D1","E5_D2","E5_D4"),
        main="unique peptides E5_b D1-D4")
#venn diagram
proteins <- as.data.frame(which(!is.na(b), arr.ind=TRUE))
D1_b <- proteins %>% dplyr::filter(col == 1)
D2_b <- proteins %>% dplyr::filter(col == 2)
D4_b <- proteins %>% dplyr::filter(col == 3)

# combine to list
lst <- list(E5_D1_b = as.double(D1_b[,1]),
            E5_D2_b = as.double(D2_b[,1]),
            E5_D4_b = as.double(D4_b[,1]))
#upset r
upset(fromList(lst), 
      order.by = "freq",
      nsets = 9,
      nintersects = 15,
      sets = c("E5_D1_b","E5_D2_b","E5_D4_b"),
      keep.order = TRUE,
      mainbar.y.label = "Number of shared proteins",
      sets.x.label = "Number of proteins per fraction",
      set_size.show = TRUE,
      set_size.angles = 0,
      set_size.scale_max = 1500
)
# venn diagram
ggvenn(lst,
       stroke_size = 0.5, set_name_size = 5, text_size = 4.25, digits = 0, fill_alpha = 0.9, label_sep = )

euler_venn_3comp(lst, label_venn = c("E5_D1_b","E5_D2_b","E5_D4_b"), color_venn = c(IG,highMW,lowMW), cex_venn = c(1,1,1), h_just = 0.5, v_just = 0.5)
```



# Compare E4 mit E5

## E4 - D1/D2 mit E5 D1-D4

```{r message=FALSE, warning=FALSE}
# D1-D2 comparison
D1_raw <- readRDS(file = "results_v002/results_run1_mqpar_extracts_2gether_E4_D1_a-j_LFQ_combined_txt_proteinGroups.txt_Unique.RDS")
D1_raw <- D1_raw[[1]]
D1 <- dplyr::select(D1_raw, contains("Pool"))
D1 <- apply(D1, 1, any)
#D1 <- as.data.frame(cbind(D1, D1_raw$FASTA))
fasta <- c()
for (i in 1:length(D1_raw$FASTA)){
  tmp <- strsplit(D1_raw$FASTA[i],split='\\|')[[1]][2]
  fasta <- c(fasta,tmp)
}
D1 <- as.data.frame(cbind(D1, fasta))
D2_raw <- readRDS(file = "results_v002/results_run1_mqpar_extracts_2gether_D2_LFQ_combined_txt_proteinGroups.txt_Unique.RDS")
D2_raw <- D2_raw[[1]]
D2 <- dplyr::select(D2_raw, contains("Gesamt"))
D2 <- apply(D2, 1, any)
fasta <- c()
for (i in 1:length(D2_raw$FASTA)){
  tmp <- strsplit(D2_raw$FASTA[i],split='\\|')[[1]][2]
  fasta <- c(fasta,tmp)
}
#D2 <- as.data.frame(cbind(D2, D2_raw$FASTA))
D2 <- as.data.frame(cbind(D2, fasta))
D1_D2_comparison  <- merge(D1, D2,
                           all = TRUE,
                           by = "fasta")
D1_D2_comparison$comb <- apply(D1_D2_comparison[,c(2:3)], 1, any)
#E5
data <- readRDS("./27_20230201_FH/results/D5_extracts_Razor.RDS")
data <- data[[1]]
b <- dplyr::select(data, contains("_b"))
b <- as.data.frame(apply(b, 1, any))
fasta <- c()
for (i in 1:length(data$FASTA)){
  tmp <- strsplit(data$FASTA[i],split='\\|')[[1]][2]
  fasta <- c(fasta,tmp)
}
b$fasta <- fasta
colnames(b) <- c("b","fasta")

#merge with D5
D1_D2_D5_comparison  <- merge(D1_D2_comparison[,c(1,4)],b,
                           all = TRUE,
                           by = "fasta")
colnames(D1_D2_D5_comparison) <- c("fasta","E4","E5")

D1_D2_D5_comparison$D1 <- as.logical(D1_D2_D5_comparison$E4)
D1_D2_D5_comparison$D2 <- as.logical(D1_D2_D5_comparison$E5)
D1_D2_D5_comparison <- D1_D2_D5_comparison[,-1]*1
D1_D2_D5_comparison[D1_D2_D5_comparison==0] <- NA

# boxplot
boxplot(sum(D1_D2_D5_comparison$E4,na.rm = TRUE),
        sum(D1_D2_D5_comparison$E5,na.rm = TRUE),
        names = c("E4_D1_D2","E5_D1_D2_D3_D4"))

proteins <- as.data.frame(which(!is.na(D1_D2_D5_comparison), arr.ind=TRUE))

E4_inter <- proteins %>% dplyr::filter(col == 1)
E5_inter <- proteins %>% dplyr::filter(col == 2)


# combine to list
lst <- list("E4_D1_D2"=as.double(E4_inter[,1]),
            "E5_D1_D2_D3_D4"=as.double(E5_inter[,1]))

upset(fromList(lst), 
      order.by = "freq",
      nsets = 10,
      nintersects = 15,
      keep.order = TRUE,
      sets=c("E4_D1_D2", "E5_D1_D2_D3_D4"),
      mainbar.y.label = "Number of shared proteins",
      sets.x.label = "Number of proteins per fraction",
      set_size.show = TRUE,
      set_size.angles = 0,
      set_size.scale_max = 1500
)

#venn digarams
#get colors
E4_col <- rgb(255,93,93, maxColorValue = 255)
E5_col <- rgb(169,208,142,maxColorValue = 255)

ggvenn(lst,
       fill_color = c(E4_col,E5_col),
       stroke_size = 0.5, set_name_size = 4, text_size = 4, digits = 0, fill_alpha = 0.9)

euler_venn_2comp(lst, label_venn = c("E4_D1_D2","E5_D1_D2_D3_D4"), color_venn = c(E4_col,E5_col), cex_venn = c(1,1,1), h_just = 0.5, v_just = 0.5)

print( c("E4_D1_D2","E5_D1_D2_D3_D4"))
sum(D1_D2_D5_comparison$E4,na.rm = TRUE)
sum(D1_D2_D5_comparison$E5,na.rm = TRUE)       
```


## E4 - D1/D2 mit E5 D1,D2 und D4

* hier wurde E5_D3 (Citratpuffer weggelassen) da es hier fast keine identifzierten proteine gabe

```{r message=FALSE, warning=FALSE}
# D1-D2 comparison
D1_raw <- readRDS(file = "results_v002/results_run1_mqpar_extracts_2gether_E4_D1_a-j_LFQ_combined_txt_proteinGroups.txt_Unique.RDS")
D1_raw <- D1_raw[[1]]
D1 <- dplyr::select(D1_raw, contains("Pool"))
D1 <- apply(D1, 1, any)
#D1 <- as.data.frame(cbind(D1, D1_raw$FASTA))
fasta <- c()
for (i in 1:length(D1_raw$FASTA)){
  tmp <- strsplit(D1_raw$FASTA[i],split='\\|')[[1]][2]
  fasta <- c(fasta,tmp)
}
D1 <- as.data.frame(cbind(D1, fasta))
D2_raw <- readRDS(file = "results_v002/results_run1_mqpar_extracts_2gether_D2_LFQ_combined_txt_proteinGroups.txt_Unique.RDS")
D2_raw <- D2_raw[[1]]
D2 <- dplyr::select(D2_raw, contains("Gesamt"))
D2 <- apply(D2, 1, any)
fasta <- c()
for (i in 1:length(D2_raw$FASTA)){
  tmp <- strsplit(D2_raw$FASTA[i],split='\\|')[[1]][2]
  fasta <- c(fasta,tmp)
}
#D2 <- as.data.frame(cbind(D2, D2_raw$FASTA))
D2 <- as.data.frame(cbind(D2, fasta))
D1_D2_comparison  <- merge(D1, D2,
                           all = TRUE,
                           by = "fasta")
D1_D2_comparison$comb <- apply(D1_D2_comparison[,c(2:3)], 1, any)
#E5
data <- readRDS("./27_20230201_FH/results/D5_extracts_Razor.RDS")
data <- data[[1]]
b <- dplyr::select(data, contains("_b"))
b <- b[,-c(5:6)]
b <- as.data.frame(apply(b, 1, any))
fasta <- c()
for (i in 1:length(data$FASTA)){
  tmp <- strsplit(data$FASTA[i],split='\\|')[[1]][2]
  fasta <- c(fasta,tmp)
}
b$fasta <- fasta
colnames(b) <- c("b","fasta")

#merge with D5
D1_D2_D5_comparison  <- merge(D1_D2_comparison[,c(1,4)],b,
                           all = TRUE,
                           by = "fasta")
colnames(D1_D2_D5_comparison) <- c("fasta","E4","E5")

D1_D2_D5_comparison$D1 <- as.logical(D1_D2_D5_comparison$E4)
D1_D2_D5_comparison$D2 <- as.logical(D1_D2_D5_comparison$E5)
D1_D2_D5_comparison <- D1_D2_D5_comparison[,-1]*1
D1_D2_D5_comparison[D1_D2_D5_comparison==0] <- NA

# boxplot
boxplot(sum(D1_D2_D5_comparison$E4,na.rm = TRUE),
        sum(D1_D2_D5_comparison$E5,na.rm = TRUE),
        names = c("E4_D1_D2","E5_D1_D2_D4"))

proteins <- as.data.frame(which(!is.na(D1_D2_D5_comparison), arr.ind=TRUE))

E4_inter <- proteins %>% dplyr::filter(col == 1)
E5_inter <- proteins %>% dplyr::filter(col == 2)


# combine to list
lst <- list("E4_D1_D2"=as.double(E4_inter[,1]),
            "E5_D1_D2_D4"=as.double(E5_inter[,1]))

upset(fromList(lst), 
      order.by = "freq",
      nsets = 10,
      nintersects = 15,
      keep.order = TRUE,
      sets=c("E4_D1_D2", "E5_D1_D2_D4"),
      mainbar.y.label = "Number of shared proteins",
      sets.x.label = "Number of proteins per fraction",
      set_size.show = TRUE,
      set_size.angles = 0,
      set_size.scale_max = 1500
)

#venn digarams
#get colors
E4_col <- rgb(255,93,93, maxColorValue = 255)
E5_col <- rgb(169,208,142,maxColorValue = 255)

ggvenn(lst,
       fill_color = c(E4_col,E5_col),
       stroke_size = 0.5, set_name_size = 4, text_size = 4, digits = 0, fill_alpha = 0.9)

euler_venn_2comp(lst, label_venn = c("E4_D1_D2","E5_D1_D2_D4"), color_venn = c(E4_col,E5_col), cex_venn = c(1,1,1), h_just = 0.5, v_just = 0.5)

print(c("E4_D1_D2","E5_D1_D2_D4"))
sum(D1_D2_D5_comparison$E4,na.rm = TRUE)
sum(D1_D2_D5_comparison$E5,na.rm = TRUE)
       
```


## E4D1 mit E5D1

```{r message=FALSE, warning=FALSE}
# E4_D1
D1_raw <- readRDS(file = "results_v002/results_run1_mqpar_extracts_2gether_E4_D1_a-j_LFQ_combined_txt_proteinGroups.txt_Unique.RDS")
D1_raw <- D1_raw[[1]]
D1 <- dplyr::select(D1_raw, contains("Pool"))
D1 <- apply(D1, 1, any)
#D1 <- as.data.frame(cbind(D1, D1_raw$FASTA))
fasta <- c()
for (i in 1:length(D1_raw$FASTA)){
  tmp <- strsplit(D1_raw$FASTA[i],split='\\|')[[1]][2]
  fasta <- c(fasta,tmp)
}
D1 <- as.data.frame(cbind(D1, fasta))

#E5
data <- readRDS("./27_20230201_FH/results/D5_extracts_Razor.RDS")
data <- data[[1]]
b <- dplyr::select(data, contains("1_b"))
b <- as.data.frame(apply(b, 1, any))
fasta <- c()
for (i in 1:length(data$FASTA)){
  tmp <- strsplit(data$FASTA[i],split='\\|')[[1]][2]
  fasta <- c(fasta,tmp)
}
b$fasta <- fasta
colnames(b) <- c("b","fasta")

#merge with D5
E4_E5_comparison  <- merge(D1,b,
                           all = TRUE,
                           by = "fasta")
colnames(E4_E5_comparison) <- c("fasta","E4","E5")

E4_E5_comparison$E4 <- as.logical(E4_E5_comparison$E4)
E4_E5_comparison$E5 <- as.logical(E4_E5_comparison$E5)
E4_E5_comparison <- E4_E5_comparison[,-1]*1
E4_E5_comparison[E4_E5_comparison==0] <- NA



# boxplot
boxplot(sum(E4_E5_comparison$E4,na.rm = TRUE),
        sum(E4_E5_comparison$E5,na.rm = TRUE),
        names = c("E4_D1","E5_b_D1"))

proteins <- as.data.frame(which(!is.na(E4_E5_comparison), arr.ind=TRUE))

E4_inter <- proteins %>% dplyr::filter(col == 1)
E5_inter <- proteins %>% dplyr::filter(col == 2)


# combine to list
lst <- list("E4_D1"=as.double(E4_inter[,1]),
            "E5_b_D1"=as.double(E5_inter[,1]))

upset(fromList(lst), 
      order.by = "freq",
      nsets = 10,
      nintersects = 15,
      keep.order = TRUE,
      sets=c("E4_D1", "E5_b_D1"),
      mainbar.y.label = "Number of shared proteins",
      sets.x.label = "Number of proteins per fraction",
      set_size.show = TRUE,
      set_size.angles = 0,
      set_size.scale_max = 1500
)

#venn digarams
#get colors
E4_col <- rgb(255,93,93, maxColorValue = 255)
E5_col <- rgb(169,208,142,maxColorValue = 255)

ggvenn(lst,
       fill_color = c(E4_col,E5_col),
       stroke_size = 0.5, set_name_size = 4, text_size = 4, digits = 0, fill_alpha = 0.9)

euler_venn_2comp(lst, label_venn = c("E4_D1","E5_b_D1"), color_venn = c(E4_col,E5_col), cex_venn = c(1,1,1), h_just = 0.5, v_just = 0.5)

print(c("E4_D1","E5_b_D1"))
sum(E4_E5_comparison$E4,na.rm = TRUE)
sum(E4_E5_comparison$E5,na.rm = TRUE)
       
```

## E4 D2 mit E5 D2 mit E5 D4

* hier wurden also 3 Gruppen verglichen

```{r message=FALSE, warning=FALSE}
# E4-D2 comparison
D2_raw <- readRDS(file = "results_v002/results_run1_mqpar_extracts_2gether_D2_LFQ_combined_txt_proteinGroups.txt_Unique.RDS")
D2_raw <- D2_raw[[1]]
D2 <- dplyr::select(D2_raw, contains("Gesamt"))
D2 <- apply(D2, 1, any)
fasta <- c()
for (i in 1:length(D2_raw$FASTA)){
  tmp <- strsplit(D2_raw$FASTA[i],split='\\|')[[1]][2]
  fasta <- c(fasta,tmp)
}
#D2 <- as.data.frame(cbind(D2, D2_raw$FASTA))
D2 <- as.data.frame(cbind(D2, fasta))
# E5-D2
data <- readRDS("./27_20230201_FH/results/D5_extracts_Razor.RDS")
data <- data[[1]]
b <- dplyr::select(data, contains("2_b"))
b <- as.data.frame(apply(b, 1, any))
fasta <- c()
for (i in 1:length(data$FASTA)){
  tmp <- strsplit(data$FASTA[i],split='\\|')[[1]][2]
  fasta <- c(fasta,tmp)
}
b$fasta <- fasta
colnames(b) <- c("b","fasta")

###merge E4_D2 pool with E5_D2
comparison  <- merge(D2,b,
                           all = TRUE,
                           by = "fasta")
colnames(comparison) <- c("fasta","E4_D2","E5_D2")

#E5_D4
data <- readRDS("./27_20230201_FH/results/D5_extracts_Razor.RDS")
data <- data[[1]]
b <- dplyr::select(data, contains("4_b"))
b <- as.data.frame(apply(b, 1, any))
fasta <- c()
for (i in 1:length(data$FASTA)){
  tmp <- strsplit(data$FASTA[i],split='\\|')[[1]][2]
  fasta <- c(fasta,tmp)
}
b$fasta <- fasta
colnames(b) <- c("b","fasta")

###merge E4_D2 poo&E5_D2 with E5_D4
comparison  <- merge(comparison,b,
                     all = TRUE,
                     by = "fasta")
colnames(comparison) <- c("fasta","E4_D2","E5_D2","E5_D4")

comparison$E4_D2 <- as.logical(comparison$E4_D2)
comparison$E5_D2 <- as.logical(comparison$E5_D2)
comparison$E5_D4 <- as.logical(comparison$E5_D4)
comparison <- comparison[,-1]*1
comparison[comparison==0] <- NA

# boxplot
boxplot(sum(comparison$E4_D2,na.rm = TRUE),
        sum(comparison$E5_D2,na.rm = TRUE),
        sum(comparison$E5_D4,na.rm = TRUE),
        names = c("E4_D2","E5_b_D2","E5_b_D4"))

proteins <- as.data.frame(which(!is.na(comparison), arr.ind=TRUE))

E4_inter <- proteins %>% dplyr::filter(col == 1)
E5_inter <- proteins %>% dplyr::filter(col == 2)
E5_4_inter <- proteins %>% dplyr::filter(col == 3)



# combine to list
lst <- list("E4_D2"=as.double(E4_inter[,1]),
            "E5_b_D2"=as.double(E5_inter[,1]),
            "E5_b_D4"=as.double(E5_4_inter[,1]))

upset(fromList(lst), 
      order.by = "freq",
      nsets = 10,
      nintersects = 15,
      keep.order = TRUE,
      sets=c("E4_D2", "E5_b_D2", "E5_b_D4"),
      mainbar.y.label = "Number of shared proteins",
      sets.x.label = "Number of proteins per fraction",
      set_size.show = TRUE,
      set_size.angles = 0,
      set_size.scale_max = 1500
)

ggvenn(lst,
       fill_color = c(IG, highMW, lowMW),
       stroke_size = 0.5, set_name_size = 4, text_size = 4, digits = 0, fill_alpha = 0.9)

euler_venn_3comp(lst, label_venn = c("E4_D2","E5_b_D2","E5_b_D4"), color_venn = c(IG, highMW, lowMW), cex_venn = c(1,1,1), h_just = 0.5, v_just = 0.5)

print(c("E4_D2","E5_b_D2","E5_b_D4"))
sum(comparison$E4_D2,na.rm = TRUE)
sum(comparison$E5_D2,na.rm = TRUE)
sum(comparison$E5_D4,na.rm = TRUE)
```



## E4 D2 mit merge(E5 D2/E5 D4)

* im vergleich zum oberen punkt wurde zuerst D2 und D4 von E5 zusammengefasst und dieses Ergebnis dann mit E4 D2 verglichen

```{r message=FALSE, warning=FALSE}
# E4-D2 comparison
D2_raw <- readRDS(file = "results_v002/results_run1_mqpar_extracts_2gether_D2_LFQ_combined_txt_proteinGroups.txt_Unique.RDS")
D2_raw <- D2_raw[[1]]
D2 <- dplyr::select(D2_raw, contains("Gesamt"))
D2 <- apply(D2, 1, any)
fasta <- c()
for (i in 1:length(D2_raw$FASTA)){
  tmp <- strsplit(D2_raw$FASTA[i],split='\\|')[[1]][2]
  fasta <- c(fasta,tmp)
}
#D2 <- as.data.frame(cbind(D2, D2_raw$FASTA))
D2 <- as.data.frame(cbind(D2, fasta))
# E5-D2
data <- readRDS("./27_20230201_FH/results/D5_extracts_Razor.RDS")
data <- data[[1]]
b <- dplyr::select(data, contains("2_b"))
b <- as.data.frame(apply(b, 1, any))
fasta <- c()
for (i in 1:length(data$FASTA)){
  tmp <- strsplit(data$FASTA[i],split='\\|')[[1]][2]
  fasta <- c(fasta,tmp)
}
b$fasta <- fasta
colnames(b) <- c("b","fasta")
###merge E4_D2 pool with E5_D2
comparison  <- merge(D2,b,
                           all = TRUE,
                           by = "fasta")
colnames(comparison) <- c("fasta","E4_D2","E5_D2")

#E5_D4
data <- readRDS("./27_20230201_FH/results/D5_extracts_Razor.RDS")
data <- data[[1]]
b <- dplyr::select(data, contains("4_b"))
b <- as.data.frame(apply(b, 1, any))
fasta <- c()
for (i in 1:length(data$FASTA)){
  tmp <- strsplit(data$FASTA[i],split='\\|')[[1]][2]
  fasta <- c(fasta,tmp)
}
b$fasta <- fasta
colnames(b) <- c("b","fasta")

###merge E4_D2 poo&E5_D2 with E5_D4
comparison  <- merge(comparison,b,
                     all = TRUE,
                     by = "fasta")
colnames(comparison) <- c("fasta","E4_D2","E5_D2","E5_D4")

comparison$comb_E5 <- apply(comparison[,c(3:4)], 1, any)


comparison$E4_D2 <- as.logical(comparison$E4_D2)
comparison$comb_E5 <- as.logical(comparison$comb_E5)
comparison <- comparison[,-c(1,3,4)]*1
comparison[comparison==0] <- NA

# boxplot
boxplot(sum(comparison$E4_D2,na.rm = TRUE),
        sum(comparison$comb_E5,na.rm = TRUE),
        names = c("E4_D2","E5_D2_D4"))

proteins <- as.data.frame(which(!is.na(comparison), arr.ind=TRUE))

E4_inter <- proteins %>% dplyr::filter(col == 1)
E5_inter <- proteins %>% dplyr::filter(col == 2)

# combine to list
lst <- list("E4_D2"=as.double(E4_inter[,1]),
            "E5_D2_D4"=as.double(E5_inter[,1]))

upset(fromList(lst), 
      order.by = "freq",
      nsets = 10,
      nintersects = 15,
      keep.order = TRUE,
      sets=c("E4_D2", "E5_D2_D4"),
      mainbar.y.label = "Number of shared proteins",
      sets.x.label = "Number of proteins per fraction",
      set_size.show = TRUE,
      set_size.angles = 0,
      set_size.scale_max = 1500
)

ggvenn(lst,
       fill_color = c(E4_col,E5_col),
       stroke_size = 0.5, set_name_size = 4, text_size = 4, digits = 0, fill_alpha = 0.9)

euler_venn_2comp(lst, label_venn = c("E4_D2","E5_D2_D4"), color_venn = c(E4_col,E5_col), cex_venn = c(1,1,1), h_just = 0.5, v_just = 0.5)

print(c("E4_D2","E5_D2_D4"))
sum(comparison$E4_D2,na.rm = TRUE)
sum(comparison$comb_E5,na.rm = TRUE)

```

# E3_D1 bzw E4_D1 und E5_D1

* bei E3 e und f genommen (je 3 fach Messung) da das mit Proteaseinhibitor ist 
* bei E4 pool das waren 2 Messungen
* bei E5 habe ich 1_b genommen - 2 messungen

```{r}
#E3_D1
D1_raw <- readRDS(file = "results_v002/results_run1_mqpar_extracts_2gether_E3_D1_LFQ_combined_txt_proteinGroups.txt_Unique.RDS")
D1_raw <- D1_raw[[1]]
D1 <- D1_raw[,c(13:18)]
D1 <- apply(D1, 1, any)
#D1 <- as.data.frame(cbind(D1, D1_raw$FASTA))
fasta <- c()
for (i in 1:length(D1_raw$FASTA)){
  tmp <- strsplit(D1_raw$FASTA[i],split='\\|')[[1]][2]
  fasta <- c(fasta,tmp)
}
E3_D1 <- as.data.frame(cbind(D1, fasta))
#E4_D1
D1_raw <- readRDS(file = "results_v002/results_run1_mqpar_extracts_2gether_E4_D1_a-j_LFQ_combined_txt_proteinGroups.txt_Unique.RDS")
D1_raw <- D1_raw[[1]]
D1 <- dplyr::select(D1_raw, contains("Pool"))
D1 <- apply(D1, 1, any)
#D1 <- as.data.frame(cbind(D1, D1_raw$FASTA))
fasta <- c()
for (i in 1:length(D1_raw$FASTA)){
  tmp <- strsplit(D1_raw$FASTA[i],split='\\|')[[1]][2]
  fasta <- c(fasta,tmp)
}
E4_D1 <- as.data.frame(cbind(D1, fasta))
#E5_D1
data_raw <- readRDS("./27_20230201_FH/results/D5_extracts_Razor.RDS")
data_raw <- data_raw[[1]]
E5_D1 <- dplyr::select(data_raw, contains("1_b"))
E5_D1 <- apply(E5_D1, 1, any)
fasta <- c()
for (i in 1:length(data_raw$FASTA)){
  tmp <- strsplit(data_raw$FASTA[i],split='\\|')[[1]][2]
  fasta <- c(fasta,tmp)
}
E5_D1 <- as.data.frame(cbind(E5_D1, fasta))

#### merging
comparison  <- merge(E3_D1, E4_D1,
                           all = TRUE,
                           by = "fasta")

comparison  <- merge(comparison, E5_D1,
                     all = TRUE,
                     by = "fasta")
colnames(comparison) <- c("fasta","E3_D1","E4_D1","E5_D1")

comparison$E3_D1 <- as.logical(comparison$E3_D1)
comparison$E4_D1 <- as.logical(comparison$E4_D1)
comparison$E5_D1 <- as.logical(comparison$E5_D1)
comparison <- comparison[,-1]*1
comparison[comparison==0] <- NA

# boxplot
boxplot(sum(comparison$E3_D1,na.rm = TRUE),
        sum(comparison$E4_D1,na.rm = TRUE),
        sum(comparison$E5_D1,na.rm = TRUE),
        names = c("E3_D1","E4_D1","E5_D1"))

proteins <- as.data.frame(which(!is.na(comparison), arr.ind=TRUE))

E3 <- proteins %>% dplyr::filter(col == 1)
E4 <- proteins %>% dplyr::filter(col == 2)
E5 <- proteins %>% dplyr::filter(col == 3)


# combine to list
lst <- list("E3"=as.double(E3[,1]),
            "E4"=as.double(E4[,1]),
            "E5"=as.double(E5[,1]))

upset(fromList(lst), 
      order.by = "freq",
      nsets = 10,
      nintersects = 15,
      keep.order = TRUE,
      sets=c("E3","E4", "E5"),
      mainbar.y.label = "Number of shared proteins",
      sets.x.label = "Number of proteins per fraction",
      set_size.show = TRUE,
      set_size.angles = 0,
      set_size.scale_max = 1500
)

ggvenn(lst,
       fill_color = c(IG,highMW,lowMW),
       stroke_size = 0.5, set_name_size = 4, text_size = 4, digits = 0, fill_alpha = 0.9)

euler_venn_3comp(lst, label_venn = c("E3_D1","E4_D1","E5_D1"), color_venn = c(IG,highMW,lowMW), cex_venn = c(1,1,1), h_just = 0.5, v_just = 0.5)

print(c("E3_D1","E4_D1","E5_D1"))
sum(comparison$E3_D1,na.rm = TRUE)
sum(comparison$E4_D1,na.rm = TRUE)
sum(comparison$E5_D1,na.rm = TRUE)


```


# Barplots SDS gels

## E4_D1

* manuell von hier abgeschrieben `file://fhwn.ac.at/TU/Forschung/wissenschaftliche%20Ver%C3%B6ffentlichungen/Publikationen/2022/Proteomics_JRS_DL/data/evaluation_summary.html#4_Compare_D1_and_D2_B1-B9_resp_B1-B10`

```{r, fig.width=4,fig.height=4}
dat <- as.data.frame(
  cbind(c(126,159,147,241,322,591,661,852,699),
        c(paste("B",1:9,sep=""))))
colnames(dat) <- c("no","fraction")
dat$no <- as.numeric(dat$no)

positions <-  c(paste("B",9:1,sep=""))


ggplot(data=dat, aes(x=fraction,y=no))+
  geom_bar(stat="identity") +
  coord_flip() +
  geom_text(aes(label=no), hjust=1.5,vjust=0.6, color="white", size=4) +
  theme(axis.title.y = element_blank(),axis.text.y = element_text(size=14)) + 
  scale_y_continuous(breaks = seq(0, 1000, by = 100)) +
  labs(y="Number of proteins per fraction") +
  scale_x_discrete(limits = positions)
```

## E4_D2

* manuell von hier abgeschrieben `file:///X:/wissenschaftliche%20Ver%C3%B6ffentlichungen/Publikationen/2022/Proteomics_JRS_DL/data/evaluation_summary_4publication.html#9_D2`

```{r, fig.width=4,fig.height=4}
dat <- as.data.frame(
  cbind(c(107,241,231,142,145,148,139,166,180,98),
        c(paste("B",1:10,sep=""))))
colnames(dat) <- c("no","fraction")
dat$no <- as.numeric(dat$no)

positions <-  c(paste("B",10:1,sep=""))

ggplot(data=dat, aes(x=fraction,y=no))+
  geom_bar(stat="identity") +
  coord_flip() +
  geom_text(aes(label=no), hjust=1.5,vjust=0.6, color="white", size=4) +
  theme(axis.title.y = element_blank(),axis.text.y = element_text(size=14)) + 
  scale_y_continuous(breaks = seq(0, 1000, by = 100)) +
  scale_x_discrete(limits = positions) +
  labs(y="Number of proteins per fraction")
```


# E3-E5 merge mit E4_D2 und E5_D2_D4

## Step1 merge(E3_e_f, E4_D1 (pool), E5_1b) mit E4_D2 E5_D2_D4

* ALLE D1 mergen (von E3_D1_e-f, E4_D1(pool) und E5_1b) und dann im Venn Diagramm mit E4_D2 und dem E5_D2_D4_merge vergleichen

```{r message=FALSE, warning=FALSE}
#E3_D1
D1_raw <- readRDS(file = "results_v002/results_run1_mqpar_extracts_2gether_E3_D1_LFQ_combined_txt_proteinGroups.txt_Unique.RDS")
D1_raw <- D1_raw[[1]]
D1 <- D1_raw[,c(13:18)]
D1 <- apply(D1, 1, any)
#D1 <- as.data.frame(cbind(D1, D1_raw$FASTA))
fasta <- c()
for (i in 1:length(D1_raw$FASTA)){
  tmp <- strsplit(D1_raw$FASTA[i],split='\\|')[[1]][2]
  fasta <- c(fasta,tmp)
}
E3_D1 <- as.data.frame(cbind(D1, fasta))
#E4_D1
D1_raw <- readRDS(file = "results_v002/results_run1_mqpar_extracts_2gether_E4_D1_a-j_LFQ_combined_txt_proteinGroups.txt_Unique.RDS")
D1_raw <- D1_raw[[1]]
D1 <- dplyr::select(D1_raw, contains("Pool"))
D1 <- apply(D1, 1, any)
#D1 <- as.data.frame(cbind(D1, D1_raw$FASTA))
fasta <- c()
for (i in 1:length(D1_raw$FASTA)){
  tmp <- strsplit(D1_raw$FASTA[i],split='\\|')[[1]][2]
  fasta <- c(fasta,tmp)
}
E4_D1 <- as.data.frame(cbind(D1, fasta))
#E5_D1
data_raw <- readRDS("./27_20230201_FH/results/D5_extracts_Razor.RDS")
data_raw <- data_raw[[1]]
E5_D1 <- dplyr::select(data_raw, contains("1_b"))
E5_D1 <- apply(E5_D1, 1, any)
fasta <- c()
for (i in 1:length(data_raw$FASTA)){
  tmp <- strsplit(data_raw$FASTA[i],split='\\|')[[1]][2]
  fasta <- c(fasta,tmp)
}
E5_D1 <- as.data.frame(cbind(E5_D1, fasta))

#### merging
comparison  <- merge(E3_D1, E4_D1,
                     all = TRUE,
                     by = "fasta")

comparison  <- merge(comparison, E5_D1,
                     all = TRUE,
                     by = "fasta")
colnames(comparison) <- c("fasta","E3_D1","E4_D1","E5_D1")

comp_D1_E3_E4_E5 <- as.data.frame(cbind(comparison$fasta,apply(comparison[,c(2:4)], 1, any)))
colnames(comp_D1_E3_E4_E5) <- c("fasta","D1")
# E4-D2 comparison
D2_raw <- readRDS(file = "results_v002/results_run1_mqpar_extracts_2gether_D2_LFQ_combined_txt_proteinGroups.txt_Unique.RDS")
D2_raw <- D2_raw[[1]]
D2 <- dplyr::select(D2_raw, contains("Gesamt"))
D2 <- apply(D2, 1, any)
fasta <- c()
for (i in 1:length(D2_raw$FASTA)){
  tmp <- strsplit(D2_raw$FASTA[i],split='\\|')[[1]][2]
  fasta <- c(fasta,tmp)
}
#D2 <- as.data.frame(cbind(D2, D2_raw$FASTA))
D2 <- as.data.frame(cbind(D2, fasta))
# E5-D2
data <- readRDS("./27_20230201_FH/results/D5_extracts_Razor.RDS")
data <- data[[1]]
b <- dplyr::select(data, contains("2_b"))
b <- as.data.frame(apply(b, 1, any))
fasta <- c()
for (i in 1:length(data$FASTA)){
  tmp <- strsplit(data$FASTA[i],split='\\|')[[1]][2]
  fasta <- c(fasta,tmp)
}
b$fasta <- fasta
colnames(b) <- c("b","fasta")
###merge E4_D2 pool with E5_D2
comparison  <- merge(D2,b,
                     all = TRUE,
                     by = "fasta")
colnames(comparison) <- c("fasta","E4_D2","E5_D2")


#E5_D4
data <- readRDS("./27_20230201_FH/results/D5_extracts_Razor.RDS")
data <- data[[1]]
b <- dplyr::select(data, contains("4_b"))
b <- as.data.frame(apply(b, 1, any))
fasta <- c()
for (i in 1:length(data$FASTA)){
  tmp <- strsplit(data$FASTA[i],split='\\|')[[1]][2]
  fasta <- c(fasta,tmp)
}
b$fasta <- fasta
colnames(b) <- c("b","fasta")

###merge E4_D2 poo&E5_D2 with E5_D4
comparison  <- merge(comparison,b,
                     all = TRUE,
                     by = "fasta")
colnames(comparison) <- c("fasta","E4_D2","E5_D2","E5_D4")

comparison$comb_E5 <- apply(comparison[,c(3:4)], 1, any)

comp_D2_E4_E5 <- as.data.frame(cbind(comparison$fasta,comparison$E4_D2,comparison$comb_E5))
colnames(comp_D2_E4_E5) <- c("fasta","E4_D2","E5_D2_D4")

comparison  <- merge(comp_D1_E3_E4_E5,comp_D2_E4_E5,
                     all = TRUE,
                     by = "fasta")
comparison$D1 <- as.logical(comparison$D1)
comparison$E4_D2 <- as.logical(comparison$E4_D2)
comparison$E5_D2_D4 <- as.logical(comparison$E5_D2_D4)
comparison <- comparison[,-1]*1
comparison[comparison==0] <- NA

# boxplot
boxplot(sum(comparison$D1,na.rm = TRUE),
        sum(comparison$E4_D2,na.rm = TRUE),
        sum(comparison$E5_D2_D4,na.rm = TRUE),
        names = c("D1","E4_D2","E5_D2_D4"))

proteins <- as.data.frame(which(!is.na(comparison), arr.ind=TRUE))

D1 <- proteins %>% dplyr::filter(col == 1)
E4_D2 <- proteins %>% dplyr::filter(col == 2)
E5_D2_D4 <- proteins %>% dplyr::filter(col == 3)


# combine to list
lst <- list("D1"=as.double(D1[,1]),
            "E4_D2"=as.double(E4_D2[,1]),
            "E5_D2_D4"=as.double(E5_D2_D4[,1]))

upset(fromList(lst), 
      order.by = "freq",
      nsets = 10,
      nintersects = 15,
      keep.order = TRUE,
      sets=c("D1","E4_D2", "E5_D2_D4"),
      mainbar.y.label = "Number of shared proteins",
      sets.x.label = "Number of proteins per fraction",
      set_size.show = TRUE,
      set_size.angles = 0,
      set_size.scale_max = 1500
)

ggvenn(lst,
       fill_color = c(IG,highMW,lowMW),
       stroke_size = 0.5, set_name_size = 4, text_size = 4, digits = 0, fill_alpha = 0.9)

euler_venn_3comp(lst, label_venn = c("D1","E4_D2","E5_b_D2_D4"), color_venn = c(IG,highMW,lowMW), cex_venn = c(1,1,1), h_just = 0.5, v_just = 0.5)

print(c("D1","E4_D2","E5_D2_D4"))
sum(comparison$D1,na.rm = TRUE)
sum(comparison$E4_D2,na.rm = TRUE)
sum(comparison$E5_D2_D4,na.rm = TRUE)

```

## Step2 merge(E3_e_f, E4_D1 a-j, E5_a, E5_b) mit E4_D2 E5_D2_D4

```{r}
aj_raw <- readRDS(file = "results_v002/results_run1_mqpar_extracts_2gether_E4_D1_a-j_LFQ_combined_txt_proteinGroups.txt_Unique.RDS")
aj_raw <- aj_raw[[1]]
aj <- aj_raw[,-ncol(aj_raw)]
FASTA <- c()
for (i in 1:length(aj_raw$FASTA)){
  tmp <- strsplit(aj_raw$FASTA[i],split='\\|')[[1]][2]
  FASTA<- c(FASTA,tmp)
}
aj <- cbind(aj,FASTA)

E3_D1_raw <- readRDS(file = "results_v002/results_run1_mqpar_extracts_2gether_E3_D1_LFQ_combined_txt_proteinGroups.txt_Unique.RDS")
E3_D1_raw <- E3_D1_raw[[1]]
E3_D1 <- E3_D1_raw[,c(13:18)]
FASTA <- c()
for (i in 1:length(E3_D1_raw$FASTA)){
  tmp <- strsplit(E3_D1_raw$FASTA[i],split='\\|')[[1]][2]
  FASTA<- c(FASTA,tmp)
}
E3_D1 <- cbind(E3_D1,FASTA)

E5_D1_raw <- readRDS(file="./27_20230201_FH/results/D5_extracts_Unique.RDS")
E5_D1_raw <- E5_D1_raw[[1]]
E5_D1 <- dplyr::select(E5_D1_raw, contains(".1"))
colnames(E5_D1) <- c(paste("Unique.peptides.","E5_D",substring(colnames(E5_D1),17),sep=""))
FASTA <- c()
for (i in 1:length(E5_D1_raw$FASTA)){
  tmp <- strsplit(E5_D1_raw$FASTA[i],split='\\|')[[1]][2]
  FASTA<- c(FASTA,tmp)
}
E5_D1 <- cbind(E5_D1,FASTA)


comp  <- merge(aj, E3_D1,
               all = TRUE,
               by = "FASTA")
comp <- merge(comp, E5_D1,
              all=TRUE,
              by="FASTA")

comp_D1 <- as.data.frame(cbind(comp$FASTA,apply(comp[,-1], 1, any)))
colnames(comp_D1) <- c("fasta","D1")

# E4-D2 comparison
D2_raw <- readRDS(file = "results_v002/results_run1_mqpar_extracts_2gether_D2_LFQ_combined_txt_proteinGroups.txt_Unique.RDS")
D2_raw <- D2_raw[[1]]
D2 <- dplyr::select(D2_raw, contains("Gesamt"))
D2 <- apply(D2, 1, any)
fasta <- c()
for (i in 1:length(D2_raw$FASTA)){
  tmp <- strsplit(D2_raw$FASTA[i],split='\\|')[[1]][2]
  fasta <- c(fasta,tmp)
}
#D2 <- as.data.frame(cbind(D2, D2_raw$FASTA))
D2 <- as.data.frame(cbind(D2, fasta))
# E5-D2
data <- readRDS("./27_20230201_FH/results/D5_extracts_Razor.RDS")
data <- data[[1]]
b <- dplyr::select(data, contains("2_b"))
b <- as.data.frame(apply(b, 1, any))
fasta <- c()
for (i in 1:length(data$FASTA)){
  tmp <- strsplit(data$FASTA[i],split='\\|')[[1]][2]
  fasta <- c(fasta,tmp)
}
b$fasta <- fasta
colnames(b) <- c("b","fasta")
###merge E4_D2 pool with E5_D2
comparison  <- merge(D2,b,
                     all = TRUE,
                     by = "fasta")
colnames(comparison) <- c("fasta","E4_D2","E5_D2")


#E5_D4
data <- readRDS("./27_20230201_FH/results/D5_extracts_Razor.RDS")
data <- data[[1]]
b <- dplyr::select(data, contains("4_b"))
b <- as.data.frame(apply(b, 1, any))
fasta <- c()
for (i in 1:length(data$FASTA)){
  tmp <- strsplit(data$FASTA[i],split='\\|')[[1]][2]
  fasta <- c(fasta,tmp)
}
b$fasta <- fasta
colnames(b) <- c("b","fasta")

###merge E4_D2 poo&E5_D2 with E5_D4
comparison  <- merge(comparison,b,
                     all = TRUE,
                     by = "fasta")
colnames(comparison) <- c("fasta","E4_D2","E5_D2","E5_D4")

comparison$comb_E5 <- apply(comparison[,c(3:4)], 1, any)

comp_D2_E4_E5 <- as.data.frame(cbind(comparison$fasta,comparison$E4_D2,comparison$comb_E5))
colnames(comp_D2_E4_E5) <- c("fasta","E4_D2","E5_D2_D4")

comparison  <- merge(comp_D1,comp_D2_E4_E5,
                     all = TRUE,
                     by = "fasta")
comparison$D1 <- as.logical(comparison$D1)
comparison$E4_D2 <- as.logical(comparison$E4_D2)
comparison$E5_D2_D4 <- as.logical(comparison$E5_D2_D4)
comparison <- comparison[,-1]*1
comparison[comparison==0] <- NA

# boxplot
boxplot(sum(comparison$D1,na.rm = TRUE),
        sum(comparison$E4_D2,na.rm = TRUE),
        sum(comparison$E5_D2_D4,na.rm = TRUE),
        names = c("D1","E4_D2","E5_D2_D4"))

proteins <- as.data.frame(which(!is.na(comparison), arr.ind=TRUE))

D1 <- proteins %>% dplyr::filter(col == 1)
E4_D2 <- proteins %>% dplyr::filter(col == 2)
E5_D2_D4 <- proteins %>% dplyr::filter(col == 3)


# combine to list
lst <- list("D1"=as.double(D1[,1]),
            "E4_D2"=as.double(E4_D2[,1]),
            "E5_D2_D4"=as.double(E5_D2_D4[,1]))

upset(fromList(lst), 
      order.by = "freq",
      nsets = 10,
      nintersects = 15,
      keep.order = TRUE,
      sets=c("D1","E4_D2", "E5_D2_D4"),
      mainbar.y.label = "Number of shared proteins",
      sets.x.label = "Number of proteins per fraction",
      set_size.show = TRUE,
      set_size.angles = 0,
      set_size.scale_max = 1500
)

ggvenn(lst,
       fill_color = c(IG,highMW,lowMW),
       stroke_size = 0.5, set_name_size = 4, text_size = 4, digits = 0, fill_alpha = 0.9)

euler_venn_3comp(lst, label_venn = c("D1","E4_D2","E5_b_D2_D4"), color_venn = c(IG,highMW,lowMW), cex_venn = c(1,1,1), h_just = 0.5, v_just = 0.5)

print(c("D1","E4_D2","E5_D2_D4"))
sum(comparison$D1,na.rm = TRUE)
sum(comparison$E4_D2,na.rm = TRUE)
sum(comparison$E5_D2_D4,na.rm = TRUE)


```


## Step3 merge(E3_e_f, E4_D1 a-j, E5_a, E5_b) mit merge(E4_D2 E5_D2_D4)

```{r message=FALSE, warning=FALSE}
aj_raw <- readRDS(file = "results_v002/results_run1_mqpar_extracts_2gether_E4_D1_a-j_LFQ_combined_txt_proteinGroups.txt_Unique.RDS")
aj_raw <- aj_raw[[1]]
aj <- aj_raw[,-ncol(aj_raw)]
FASTA <- c()
for (i in 1:length(aj_raw$FASTA)){
  tmp <- strsplit(aj_raw$FASTA[i],split='\\|')[[1]][2]
  FASTA<- c(FASTA,tmp)
}
aj <- cbind(aj,FASTA)

E3_D1_raw <- readRDS(file = "results_v002/results_run1_mqpar_extracts_2gether_E3_D1_LFQ_combined_txt_proteinGroups.txt_Unique.RDS")
E3_D1_raw <- E3_D1_raw[[1]]
E3_D1 <- E3_D1_raw[,c(13:18)]
FASTA <- c()
for (i in 1:length(E3_D1_raw$FASTA)){
  tmp <- strsplit(E3_D1_raw$FASTA[i],split='\\|')[[1]][2]
  FASTA<- c(FASTA,tmp)
}
E3_D1 <- cbind(E3_D1,FASTA)

E5_D1_raw <- readRDS(file="./27_20230201_FH/results/D5_extracts_Unique.RDS")
E5_D1_raw <- E5_D1_raw[[1]]
E5_D1 <- dplyr::select(E5_D1_raw, contains(".1"))
colnames(E5_D1) <- c(paste("Unique.peptides.","E5_D",substring(colnames(E5_D1),17),sep=""))
FASTA <- c()
for (i in 1:length(E5_D1_raw$FASTA)){
  tmp <- strsplit(E5_D1_raw$FASTA[i],split='\\|')[[1]][2]
  FASTA<- c(FASTA,tmp)
}
E5_D1 <- cbind(E5_D1,FASTA)


comp  <- merge(aj, E3_D1,
               all = TRUE,
               by = "FASTA")
comp <- merge(comp, E5_D1,
              all=TRUE,
              by="FASTA")

comp_D1 <- as.data.frame(cbind(comp$FASTA,apply(comp[,-1], 1, any)))
colnames(comp_D1) <- c("fasta","D1")

# E4-D2 comparison
D2_raw <- readRDS(file = "results_v002/results_run1_mqpar_extracts_2gether_D2_LFQ_combined_txt_proteinGroups.txt_Unique.RDS")
D2_raw <- D2_raw[[1]]
D2 <- dplyr::select(D2_raw, contains("Gesamt"))
D2 <- apply(D2, 1, any)
fasta <- c()
for (i in 1:length(D2_raw$FASTA)){
  tmp <- strsplit(D2_raw$FASTA[i],split='\\|')[[1]][2]
  fasta <- c(fasta,tmp)
}
#D2 <- as.data.frame(cbind(D2, D2_raw$FASTA))
D2 <- as.data.frame(cbind(D2, fasta))
# E5-D2
data <- readRDS("./27_20230201_FH/results/D5_extracts_Razor.RDS")
data <- data[[1]]
b <- dplyr::select(data, contains("2_b"))
b <- as.data.frame(apply(b, 1, any))
fasta <- c()
for (i in 1:length(data$FASTA)){
  tmp <- strsplit(data$FASTA[i],split='\\|')[[1]][2]
  fasta <- c(fasta,tmp)
}
b$fasta <- fasta
colnames(b) <- c("b","fasta")
###merge E4_D2 pool with E5_D2
comparison  <- merge(D2,b,
                     all = TRUE,
                     by = "fasta")
colnames(comparison) <- c("fasta","E4_D2","E5_D2")


#E5_D4
data <- readRDS("./27_20230201_FH/results/D5_extracts_Razor.RDS")
data <- data[[1]]
b <- dplyr::select(data, contains("4_b"))
b <- as.data.frame(apply(b, 1, any))
fasta <- c()
for (i in 1:length(data$FASTA)){
  tmp <- strsplit(data$FASTA[i],split='\\|')[[1]][2]
  fasta <- c(fasta,tmp)
}
b$fasta <- fasta
colnames(b) <- c("b","fasta")

###merge E4_D2 poo&E5_D2 with E5_D4
comparison  <- merge(comparison,b,
                     all = TRUE,
                     by = "fasta")
colnames(comparison) <- c("fasta","E4_D2","E5_D2","E5_D4")

comparison$comb_E5 <- apply(comparison[,c(3:4)], 1, any)

comp_D2_E4_E5 <- as.data.frame(cbind(comparison$fasta,comparison$E4_D2,comparison$comb_E5))
colnames(comp_D2_E4_E5) <- c("fasta","E4_D2","E5_D2_D4")

comparison  <- merge(comp_D1,comp_D2_E4_E5,
                     all = TRUE,
                     by = "fasta")

comparison$D2_D4 <- apply(comparison[,c(3:4)], 1, any)

comparison$D1 <- as.logical(comparison$D1)
comparison$D2_D4 <- as.logical(comparison$D2_D4)
comparison <- comparison[,-c(1,3,4)]*1
comparison[comparison==0] <- NA

# boxplot
boxplot(sum(comparison$D1,na.rm = TRUE),
        sum(comparison$D2_D4,na.rm = TRUE),
        names = c("D1","D2_D4"))

proteins <- as.data.frame(which(!is.na(comparison), arr.ind=TRUE))

D1 <- proteins %>% dplyr::filter(col == 1)
D2_D4 <- proteins %>% dplyr::filter(col == 2)


# combine to list
lst <- list("D1"=as.double(D1[,1]),
            "D2_D4"=as.double(D2_D4[,1]))

upset(fromList(lst), 
      order.by = "freq",
      nsets = 10,
      nintersects = 15,
      keep.order = TRUE,
      sets=c("D1","D2_D4"),
      mainbar.y.label = "Number of shared proteins",
      sets.x.label = "Number of proteins per fraction",
      set_size.show = TRUE,
      set_size.angles = 0,
      set_size.scale_max = 1500
)

ggvenn(lst,
       fill_color = c(IG,highMW,lowMW),
       stroke_size = 0.5, set_name_size = 4, text_size = 4, digits = 0, fill_alpha = 0.9)

euler_venn_3comp(lst, label_venn = c("D1","D2_D4"), color_venn = c(IG,highMW), cex_venn = c(1,1,1), h_just = 0.5, v_just = 0.5)

print(c("D1","D2_D4"))
sum(comparison$D1,na.rm = TRUE)
sum(comparison$D2_D4,na.rm = TRUE)


```


# Basic information

## Auswertung Extrakte

-   alle extrakte wurden am server ausgewertet (mqpar files:
    `/proj/proteomics/100_extracts_2gether/mqpar`)
-   es wurden thematisch ähnliche extrakte gemeinsam ausgewertet
    (farblich hier markiert
    `\\fhwn.ac.at\TU\Forschung\wissenschaftliche Veröffentlichungen\Publikationen\2022\Proteomics_JRS_DL\data\Extraktionen_gesamt_V2_DL.xlsx`)
    -   D2
    -   E3_D1
    -   E4_D1_AF_TF
    -   E4_D1_a\_j
    -   E4_D1_B
    -   E5 (liegt unter data/27_202301_FH)
-   `X:\wissenschaftliche Veröffentlichungen\Publikationen\2022\Proteomics_JRS_DL\data\proteinGroups`:
    enthalten die proteinGroups.txt files die mit Razor/Unique
    ausgewertet wurden
-   später wurde dann zum mqpar file auch LFQ hinzugefügt und neu
    ausgewertet; diese Ergebnisse werden auch fürs Paper verwendet und
    sind in `proteinGroups_v002` zu finden

### Basic postprocessing

-   die proteingroups.txt files wurden dann mit
    `basis-postprocessing.R`ausgewertet
-   die Ergebnisse finden sich `results` bzw `results_v002`
    -   es wurden RDS, csv und html files abgespeichert
    -   diese Files wurden dann für die weitere Datenaufarbeitung
        verwendet
    -   es wurden 2 pepitdes verwendet

### MBR

-   die Prozessierung erfolgte 1x ohne und 1x mit MBR
-   zb
    `X:/wissenschaftliche Veröffentlichungen/Publikationen/2022/Proteomics_JRS_DL/data/results_v002`
    -   `results_run1_mqpar_extracts_2gether_D2_LFQ_combined_txt_proteinGroups.txt_LFQ.RDS`:
        hier wurde D2 mit Unique/Razor/LFQ im xml file ausgewertet und
        die LFQ Ergebnisse abgespeichert
    -   `results_run1_mqpar_extracts_2gether_D2_MBR_LFQ_combined_txt_proteinGroups.txt_LFQ.RDS`:
        das gleiche wie oben nur mit MBR im xml file
-   als weiteres Beispiel für die Dateinamen
    -   `results_run1_mqpar_extracts_2gether_D2_MBR_LFQ_combined_txt_proteinGroups.txt_Razor.RDS`:
        hier wurde D2 mit Unique/Razor/LFQ ausgewertet und die Razor
        Ergebnisse abgespeichert

### files im results_v002 ordner

-   hier sind die Ergebnisse der LFQ, Razor und Unique als RDS datei
    abgespeichert die dann für weitere Auswertungen verwendet werden
-   zusätzlich sind die html files die automatisch generiert wurden
    dabie
-   die csv datei enthält die gesamtidentifizierten/Quantifzierten
    proteine

## TA/YEAST

-   folder `X:\wissenschaftliche Veröffentlichungen\Publikationen\2022\Proteomics_JRS_DL\data\ta_yeast`
-   die Mehlwürmer wurden in Weizenmehl/Hefeextrakt gezüchtet
-   hier wurde getestet ob in den Extrakten auch
    Weizenproteine/Hefeproteine gefunden werden
-   zusätzlich wurde gegen Rattenproteine bzw A.thaliana proteine
    gesucht um zu sehen ob komplett fremde organismenproteine gefunden
    werden können
-   genommen wurden die mqpar files zur erzeugugn von proteinGroups,
    also ohne LFQ und nur Razor/Unique
-   Ergebnisfile/Zusammenfassungfile: summary-ta-yeast.Rmd
    -   es zeigt sich, dass sowohol Weizen/Hefe als auch Ratten- und a
        thaliana Proteine gefunden werden können
    -   Ausmaß max. 8% der Gesamtproteine

## evaluation_proteinmasses.html

-   hier wurden die Proteinmassen von B1-B9 E4_D1_B bzw E4_D2
    automatisch von uniprot herunterladen und visualisert um zu sehen ob
    sich die proteinmassen je fraktion unterschdein -\> conlusio: gerade
    in den ersten fraktionen ziemlich gleiche masse; erst gegen ende
    werden die proteine merklich kleiner
-   weiters wurden die Massen der geteilten Proteine in E4_D1_B
    angeschaut
    -   das waren ca. 70 Proteine die in allen Fraktionen (B1-B9)
        vorgekommen sind

## files evaluation summary ...

-   hier wurden die RDS files kombiniert eingelesen und mit den
    verschiedenen parametern diagramme erstellt um versch. einstellungen
    während der prozessierung in maxquant zu vergleichen
    -   zb wurden Unique/Razor/LFQ mit MBR und ohne MBR verglichen zb
        `file:///X:/wissenschaftliche%20Ver%C3%B6ffentlichungen/Publikationen/2022/Proteomics_JRS_DL/data/evaluation_summary_unique.html`
        bzw.
        `file:///X:/wissenschaftliche%20Ver%C3%B6ffentlichungen/Publikationen/2022/Proteomics_JRS_DL/data/evaluation_summary_MBR_unique.html`
    -   zusätzlich wurden noch versch imputation methods
        ausprobiert\^`file:///X:/wissenschaftliche%20Ver%C3%B6ffentlichungen/Publikationen/2022/Proteomics_JRS_DL/data/evaluation_summary_unique_imputed_2consecutivevalues.html`
        oder
        `file:///X:/wissenschaftliche%20Ver%C3%B6ffentlichungen/Publikationen/2022/Proteomics_JRS_DL/data/evaluation_summary_unique_imputed.html`
-   die zusammengefassten ergebnisse finden sich hier
    -   `file:///X:/wissenschaftliche%20Ver%C3%B6ffentlichungen/Publikationen/2022/Proteomics_JRS_DL/data/evaluation_summary.html`

## merging of 2 data sets

-   bei der auswertung mussten teilweise datensätze gemerged werden
-   zb wenn D2 und D1 verglichen werden sollen, wurden D1 und D2 ja in
    MaxQuant separat ausgewertet weswegen es 2 proteinGroupstxt files
    und auch ergebnisfiles (RDS files) gibt
-   zum mergen wurde die spalte `fasta.headers` des proteingroups txt
    files verwendet
-   im 1. Schritt wurde der gesamte text dieser spalte verwendet
-   im 2. und hier dargestellten schritte wurde die accession number des
    ersten proteins der jeweiligen protein group mit folgendem code extrahiert und zum mergen
    verwendet

```{=html}
<!-- -->
```
    fasta <- c()
    for (i in 1:length(AF_TF_raw$FASTA)){
      tmp <- strsplit(AF_TF_raw$FASTA[i],split='\\|')[[1]][2]
      fasta <- c(fasta,tmp)
    }
