---
output:
  html_document:
    df_print: paged
    fig.align: center
    self_contained: yes 
    fig.height: 4
    fig.width: 8
    theme: united
    toc: yes
    toc_depth: 4
    toc_float: yes
    number_sections: yes
    code_folding: hide
title: "Auswertung TC Daten"
author: "David Lilek"
date: "`r format(Sys.time(), '%d %B %Y, %X')`"
editor_options: 
  markdown: 
    wrap: 72
---

# Basic information

## Auswertung Extrakte

-   alle extrakte wurden am server ausgewertet (mqpar files:
    `/proj/proteomics/100_extracts_2gether/mqpar`)
-   es wurden thematisch ähnliche extrakte gemeinsam ausgewertet
    (farblich hier markiert
    `\\fhwn.ac.at\TU\Forschung\wissenschaftliche Veröffentlichungen\Publikationen\2022\Proteomics_JRS_DL\data\Extraktionen_gesamt_V2_DL.xlsx`)
    -   D2
    -   E3_D1
    -   E4_D1_AF_TF
    -   E4_D1_a\_j
    -   E4_D1_B
    -   E5 (liegt unter data/27_202301_FH)
-   `X:\wissenschaftliche Veröffentlichungen\Publikationen\2022\Proteomics_JRS_DL\data\proteinGroups`:
    enthalten die proteinGroups.txt files die mit Razor/Unique
    ausgewertet wurden
-   später wurde dann zum mqpar file auch LFQ hinzugefügt und neu
    ausgewertet; diese Ergebnisse werden auch fürs Paper verwendet und
    sind in `proteinGroups_v002` zu finden

### Basic postprocessing

-   die proteingroups.txt files wurden dann mit
    `basis-postprocessing.R`ausgewertet
-   die Ergebnisse finden sich `results` bzw `results_v002`
    -   es wurden RDS, csv und html files abgespeichert
    -   diese Files wurden dann für die weitere Datenaufarbeitung
        verwendet
    -   es wurden 2 pepitdes verwendet

### MBR

-   die Prozessierung erfolgte 1x ohne und 1x mit MBR
-   zb
    `X:/wissenschaftliche Veröffentlichungen/Publikationen/2022/Proteomics_JRS_DL/data/results_v002`
    -   `results_run1_mqpar_extracts_2gether_D2_LFQ_combined_txt_proteinGroups.txt_LFQ.RDS`:
        hier wurde D2 mit Unique/Razor/LFQ im xml file ausgewertet und
        die LFQ Ergebnisse abgespeichert
    -   `results_run1_mqpar_extracts_2gether_D2_MBR_LFQ_combined_txt_proteinGroups.txt_LFQ.RDS`:
        das gleiche wie oben nur mit MBR im xml file
-   als weiteres Beispiel für die Dateinamen
    -   `results_run1_mqpar_extracts_2gether_D2_MBR_LFQ_combined_txt_proteinGroups.txt_Razor.RDS`:
        hier wurde D2 mit Unique/Razor/LFQ ausgewertet und die Razor
        Ergebnisse abgespeichert

### files im results_v002 ordner

-   hier sind die Ergebnisse der LFQ, Razor und Unique als RDS datei
    abgespeichert die dann für weitere Auswertungen verwendet werden
-   zusätzlich sind die html files die automatisch generiert wurden
    dabie
-   die csv datei enthält die gesamtidentifizierten/Quantifzierten
    proteine

## TA/YEAST

-   folder `X:\wissenschaftliche Veröffentlichungen\Publikationen\2022\Proteomics_JRS_DL\data\ta_yeast`
-   die Mehlwürmer wurden in Weizenmehl/Hefeextrakt gezüchtet
-   hier wurde getestet ob in den Extrakten auch
    Weizenproteine/Hefeproteine gefunden werden
-   zusätzlich wurde gegen Rattenproteine bzw A.thaliana proteine
    gesucht um zu sehen ob komplett fremde organismenproteine gefunden
    werden können
-   genommen wurden die mqpar files zur erzeugugn von proteinGroups,
    also ohne LFQ und nur Razor/Unique
-   Ergebnisfile/Zusammenfassungfile: summary-ta-yeast.Rmd
    -   es zeigt sich, dass sowohol Weizen/Hefe als auch Ratten- und a
        thaliana Proteine gefunden werden können
    -   Ausmaß max. 8% der Gesamtproteine

## evaluation_proteinmasses.html

-   hier wurden die Proteinmassen von B1-B9 E4_D1_B bzw E4_D2
    automatisch von uniprot herunterladen und visualisert um zu sehen ob
    sich die proteinmassen je fraktion unterschdein -\> conlusio: gerade
    in den ersten fraktionen ziemlich gleiche masse; erst gegen ende
    werden die proteine merklich kleiner
-   weiters wurden die Massen der geteilten Proteine in E4_D1_B
    angeschaut
    -   das waren ca. 70 Proteine die in allen Fraktionen (B1-B9)
        vorgekommen sind

## files evaluation summary ...

-   hier wurden die RDS files kombiniert eingelesen und mit den
    verschiedenen parametern diagramme erstellt um versch. einstellungen
    während der prozessierung in maxquant zu vergleichen
    -   zb wurden Unique/Razor/LFQ mit MBR und ohne MBR verglichen zb
        `file:///X:/wissenschaftliche%20Ver%C3%B6ffentlichungen/Publikationen/2022/Proteomics_JRS_DL/data/evaluation_summary_unique.html`
        bzw.
        `file:///X:/wissenschaftliche%20Ver%C3%B6ffentlichungen/Publikationen/2022/Proteomics_JRS_DL/data/evaluation_summary_MBR_unique.html`
    -   zusätzlich wurden noch versch imputation methods
        ausprobiert\^`file:///X:/wissenschaftliche%20Ver%C3%B6ffentlichungen/Publikationen/2022/Proteomics_JRS_DL/data/evaluation_summary_unique_imputed_2consecutivevalues.html`
        oder
        `file:///X:/wissenschaftliche%20Ver%C3%B6ffentlichungen/Publikationen/2022/Proteomics_JRS_DL/data/evaluation_summary_unique_imputed.html`
-   die zusammengefassten ergebnisse finden sich hier
    -   `file:///X:/wissenschaftliche%20Ver%C3%B6ffentlichungen/Publikationen/2022/Proteomics_JRS_DL/data/evaluation_summary.html`

## merging of 2 data sets

-   bei der auswertung mussten teilweise datensätze gemerged werden
-   zb wenn D2 und D1 verglichen werden sollen, wurden D1 und D2 ja in
    MaxQuant separat ausgewertet weswegen es 2 proteinGroupstxt files
    und auch ergebnisfiles (RDS files) gibt
-   zum mergen wurde die spalte `fasta.headers` des proteingroups txt
    files verwendet
-   im 1. Schritt wurde der gesamte text dieser spalte verwendet
-   im 2. und hier dargestellten schritte wurde die accession number des
    ersten proteins der jeweiligen protein group mit folgendem code extrahiert und zum mergen
    verwendet

```{=html}
<!-- -->
```
    fasta <- c()
    for (i in 1:length(AF_TF_raw$FASTA)){
      tmp <- strsplit(AF_TF_raw$FASTA[i],split='\\|')[[1]][2]
      fasta <- c(fasta,tmp)
    }

# Besprechungen

## Besprechung 230308

wieso finden wir immer gleich große proteine obwohl wir sie eigentlich
getrennt haben? wie erklären wir, dass gewisse proteine über alle
fraktionen kommen? -\> die konzentration der proteine ändert sich nicht
wirklich über die fraktionen
`file:///X:/wissenschaftliche%20Ver%C3%B6ffentlichungen/Publikationen/2022/Proteomics_JRS_DL/data/evaluation_proteinmasses.html#31_SlopeR2`

-   Vergleich Puffer

-   PI funktioniert (E3_D1)

-   AF/TF/B

-   versch. konzentrationen B20,B30,B40 -\> weglassen da wir wie in
    literatur ja eh bei b20 die besten ergebnisse hatten

-   zuerst pool und dann fraktionierungen D1/D2 und ev. 3 und 4.
    dimension wenn es passt

-   poolD1 und poolD2 vergleichen wie in grafik links unten im word

### todo

-   proteinquantifizierungen aller D1 also E4_D1,E5_D1,... suchen und
    mit der proteinanzahl abgleichen
    -   -\> gibt es korrelationen zwischen konzentration protein
        nasschemisch und anzahl proteine
-   ideas venn diagram
    `https://cran.r-project.org/web/packages/eulerr/vignettes/gallery.html`
* judith fragen: nehmen wir für B - also ingel digest - E4_D1_Pool_G_B1-B9 oder E4_D2_Pool_20_B1-B9?
  + bis jetzt immer G; es macht aber wie mann bei punkt `8 Compare D1-D2
8.1 fractions` sieht keinen großen unterschied
  + bzw `compare E4_D1_Pool_G_B1-B9 and E4_D1_Pool_20_B1-B9`
* besprechen e5_a bzw b D1-D3 bzw D1-D4

# Load Libraries

```{r message=FALSE, warning=FALSE}
library(ggplot2)
library(tidyverse)
library(dplyr)
library(UpSetR)
library(pheatmap)
library(ggvenn)
library(knitr)
library(reshape2)
library(eulerr)
#library(venneuler)
#library(VennDiagram) 
```

# Comparison E3_D1 - noTreatment, Heatshock, ProteaseInhibitor

-   E3_D1 no Treatment X; Heatshok (HS); Protease Inhibitor (PI) -\>
    outcome: mit PI die meisten Identifikationen gefunden -\> deswegen
    E4 und E5 mit PI durchgeführt

-   extraktionsreplikate

    -   a/b no treatment
    -   c/d heatshock
    -   e/f protease inhibitor

-   3 mittelwerte machen und mittels varianzanalyse unterschiede testen
    `file:///X:/wissenschaftliche%20Ver%C3%B6ffentlichungen/Publikationen/2022/Proteomics_JRS_DL/data/results_v002/results_run1_mqpar_extracts_2gether_E3_D1_LFQ_combined_txt_proteinGroups.txt_post-processing.html#52_Summary_Unique`

-   varianzanalyse mit anschließendem tukey hsd test

-   es unterscheiden sich alle treatments signifikant voneinander

```{r}
data <- read.csv("Extraktionen_gesamt_V2_DL.csv", sep=";")
data_sub <- as.data.frame(data$Unique.no.MBR.2.peptides)
data_sub <- as.data.frame(data_sub[c(1:18),])
data_sub$group <- c(rep("noTreatment",6),
                   rep("Heatshock",6),
                   rep("ProteaseInhibitor",6))
colnames(data_sub) <- c("number","group")
ggplot(data_sub, aes(x=group, y=number)) + 
  geom_boxplot()
anova <- aov(number ~ group, data=data_sub)
summary(anova)
Tukey <- TukeyHSD(anova)
plot(Tukey)
Tukey
```

# D1 - B - AF - TF

```{r}

############## B
data <- readRDS(file = "results_v002/results_run1_mqpar_extracts_2gether_E4_D1_B_LFQ_combined_txt_proteinGroups.txt_Unique.RDS")
data <- data[[1]]
data_sub <- dplyr::select(data, contains("Pool_G"))
#get rid of FALSE values
B <- data_sub*1
B[B==0] <- NA
ind <- apply(B, 1, function(x) all(is.na(x)))


############AF
data <- readRDS(file = "results_v002/results_run1_mqpar_extracts_2gether_E4_D1_AF_TF_LFQ_combined_txt_proteinGroups.txt_Unique.RDS")
data <- data[[1]]
data_sub <- dplyr::select(data, contains("Pool_AF"))
#get rid of FALSE values
AF <- data_sub*1
AF[AF==0] <- NA
ind <- apply(AF, 1, function(x) all(is.na(x)))

############TF
data <- readRDS(file = "results_v002/results_run1_mqpar_extracts_2gether_E4_D1_AF_TF_LFQ_combined_txt_proteinGroups.txt_Unique.RDS")
data <- data[[1]]
data_sub <- dplyr::select(data, contains("Pool_TF"))
#get rid of FALSE values
TF <- data_sub*1
TF[TF==0] <- NA
ind <- apply(TF, 1, function(x) all(is.na(x)))


## compare AF TF B poster
AF_TF_raw <- readRDS(file = "results_v002/results_run1_mqpar_extracts_2gether_E4_D1_AF_TF_LFQ_combined_txt_proteinGroups.txt_Unique.RDS")
AF_TF_raw <- AF_TF_raw[[1]]
AF <- dplyr::select(AF_TF_raw, contains("AF"))
AF <- apply(AF, 1, any)
fasta <- c()
for (i in 1:length(AF_TF_raw$FASTA)){
  tmp <- strsplit(AF_TF_raw$FASTA[i],split='\\|')[[1]][2]
  fasta <- c(fasta,tmp)
}
AF <- as.data.frame(cbind(AF, fasta))
TF <- dplyr::select(AF_TF_raw, contains("TF"))
TF <- apply(TF, 1, any)
TF <- as.data.frame(cbind(TF, fasta))
B_raw <- readRDS(file = "results_v002/results_run1_mqpar_extracts_2gether_E4_D1_B_LFQ_combined_txt_proteinGroups.txt_Unique.RDS")
B_raw <- B_raw[[1]]
B <- dplyr::select(B_raw, contains("Pool_G"))
B <- apply(B, 1, any)
fasta <- c()
for (i in 1:length(B_raw$FASTA)){
  tmp <- strsplit(B_raw$FASTA[i],split='\\|')[[1]][2]
  fasta <- c(fasta,tmp)
}
B <- as.data.frame(cbind(B, fasta))

AF_TF_B_comparison  <- merge(B, AF,
                           all = TRUE,
                           by = "fasta")
AF_TF_B_comparison <- merge(AF_TF_B_comparison, TF,
                            all = TRUE,
                            by = "fasta")

AF_TF_B_comparison$B <- as.logical(AF_TF_B_comparison$B)
AF_TF_B_comparison$AF <- as.logical(AF_TF_B_comparison$AF)
AF_TF_B_comparison$TF <- as.logical(AF_TF_B_comparison$TF)
AF_TF_B_comparison <- AF_TF_B_comparison[,-1]*1
AF_TF_B_comparison[AF_TF_B_comparison==0] <- NA

# boxplot
boxplot(sum(AF_TF_B_comparison$B,na.rm = TRUE),
        sum(AF_TF_B_comparison$AF,na.rm = TRUE),
        sum(AF_TF_B_comparison$TF,na.rm = TRUE),
        names = c("B","AF","TF"))

proteins <- as.data.frame(which(!is.na(AF_TF_B_comparison), arr.ind=TRUE))

B <- proteins %>% dplyr::filter(col == 1)
AF <- proteins %>% dplyr::filter(col == 2)
TF <- proteins %>% dplyr::filter(col == 3)


# combine to list
lst <- list("B"=as.double(B[,1]),
            "AF"=as.double(AF[,1]),
            "TF"=as.double(TF[,1]))

upset(fromList(lst), 
      order.by = "freq",
      nsets = 10,
      nintersects = 15,
      keep.order = TRUE,
      sets=c("B", "AF", "TF"),
      mainbar.y.label = "Number of shared proteins",
      sets.x.label = "Number of proteins per fraction",
      set_size.show = TRUE,
      set_size.angles = 0,
      set_size.scale_max = 1800
)


#get colors
#total protein extract
total <- rgb(255,93,93, maxColorValue = 255)
#fractions high MW SEC
highMW <- rgb(169,208,142,maxColorValue = 255)
#fractions low MW SEC
lowMW <- rgb(255,217,102, maxColorValue = 255)
#in-gel digest
IG <- rgb(180,198,231,maxColorValue = 255)

ggvenn(lst,
       fill_color = c(IG,lowMW,highMW),
       stroke_size = 0.5, set_name_size = 5, text_size = 4.25, digits = 0, fill_alpha = 0.9, label_sep = )
#ggsave("./pics/unique_AF_TF_B.png")

#compare B to pool from 0609 
B_raw <- readRDS(file = "results_v002/results_run1_mqpar_extracts_2gether_E4_D1_B_LFQ_combined_txt_proteinGroups.txt_Unique.RDS")
B_raw <- B_raw[[1]]
B <- dplyr::select(B_raw, contains("Pool_G"))
B <- apply(B, 1, any)
fasta <- c()
for (i in 1:length(B_raw$FASTA)){
  tmp <- strsplit(B_raw$FASTA[i],split='\\|')[[1]][2]
  fasta <- c(fasta,tmp)
}
B <- as.data.frame(cbind(B, fasta))
aj_raw <- readRDS(file = "results_v002/results_run1_mqpar_extracts_2gether_E4_D1_a-j_LFQ_combined_txt_proteinGroups.txt_Unique.RDS")
aj_raw <- aj_raw[[1]]
aj <- dplyr::select(aj_raw, contains("Pool"))
aj <- apply(aj, 1, any)
fasta <- c()
for (i in 1:length(aj_raw$FASTA)){
  tmp <- strsplit(aj_raw$FASTA[i],split='\\|')[[1]][2]
  fasta <- c(fasta,tmp)
}
aj <- as.data.frame(cbind(aj, fasta))
B_aj_comparison  <- merge(B, aj,
                             all = TRUE,
                             by = "fasta")

B_aj_comparison$B <- as.logical(B_aj_comparison$B)
B_aj_comparison$aj <- as.logical(B_aj_comparison$aj)
B_aj_comparison <- B_aj_comparison[,-1]*1
B_aj_comparison[B_aj_comparison==0] <- NA
boxplot(sum(B_aj_comparison$B,na.rm = TRUE),
        sum(B_aj_comparison$aj,na.rm = TRUE),
        names = c("B","E4_D1_pool"))

proteins <- as.data.frame(which(!is.na(B_aj_comparison), arr.ind=TRUE))

B_inter <- proteins %>% dplyr::filter(col == 1)
aj_inter <- proteins %>% dplyr::filter(col == 2)

# combine to list
lst <- list("B"=as.double(B_inter[,1]),
            "E4_D1_pool"=as.double(aj_inter[,1]))

upset(fromList(lst), 
      order.by = "freq",
      nsets = 10,
      nintersects = 15,
      keep.order = TRUE,
      sets=c("B", "E4_D1_pool"),
      mainbar.y.label = "Number of shared proteins",
      sets.x.label = "Number of proteins per fraction",
      set_size.show = TRUE,
      set_size.angles = 0,
      set_size.scale_max = 1800
)

ggvenn(lst,
       stroke_size = 0.5, set_name_size = 5, text_size = 4, digits = 0, fill_alpha = 0.9)
#ggsave("./pics/unique_B_D1pool.png")

plot(venn(lst))
plot(euler(lst, shape = "ellipse"), quantities = TRUE)


```


## compare E4_D1_Pool_G_B1-B9 and E4_D1_Pool_20_B1-B9

* um Unterschiede zwischen B1-B9 in Pool_G und Pool_20 zu sehen
* habe mir die Proteine auch angeschaut die unterschiedlich sind. 31 davon sind uncharacterized proteins
* teilweise sind sich die proteine schon ähnlich zumindest von der accesion number

```{r}
#E4_D1_Pool_G_B1-B9
data <- readRDS(file = "results_v002/results_run1_mqpar_extracts_2gether_E4_D1_B_LFQ_combined_txt_proteinGroups.txt_Unique.RDS")
data <- data[[1]]
data_sub <- dplyr::select(data, contains("Pool_G"))
B_D1<-apply(data_sub, 1, any)
fasta <- c()
for (i in 1:length(data$FASTA)){
  tmp <- strsplit(data$FASTA[i],split='\\|')[[1]][2]
  fasta <- c(fasta,tmp)
}
B_D1 <- as.data.frame(cbind(B_D1, fasta))
#B_D1 <- as.data.frame(cbind(B_D1, data$FASTA))

#E4_D1_Pool_20_B1-B9

D1_raw <- readRDS(file = "results_v002/results_run1_mqpar_extracts_2gether_E4_D1_B_LFQ_combined_txt_proteinGroups.txt_Unique.RDS")
D1_raw <- D1_raw[[1]]
D1 <- dplyr::select(D1_raw, contains("20"))
D1 <- apply(D1, 1, any)
#D1 <- as.data.frame(cbind(D1, D1_raw$FASTA))
fasta <- c()
for (i in 1:length(data$FASTA)){
  tmp <- strsplit(data$FASTA[i],split='\\|')[[1]][2]
  fasta <- c(fasta,tmp)
}
D1 <- as.data.frame(cbind(D1, fasta))

G_pool_comparison  <- merge(B_D1, D1,
                           all = TRUE,
                           by = "fasta")

G_pool_comparison$B_D1 <- as.logical(G_pool_comparison$B_D1)
G_pool_comparison$D1 <- as.logical(G_pool_comparison$D1)
G_pool_comparison <- G_pool_comparison[,-1]*1
G_pool_comparison[G_pool_comparison==0] <- NA

# boxplot
boxplot(sum(G_pool_comparison$B_D1,na.rm = TRUE),
        sum(G_pool_comparison$D1,na.rm = TRUE),
        names = c("pool_G","pool_20"))

proteins <- as.data.frame(which(!is.na(G_pool_comparison), arr.ind=TRUE))

B_D1_inter <- proteins %>% dplyr::filter(col == 1)
D1_inter <- proteins %>% dplyr::filter(col == 2)

# combine to list
lst <- list("pool_G"=as.double(B_D1_inter[,1]),
            "pool_20"=as.double(D1_inter[,1]))

upset(fromList(lst), 
      order.by = "freq",
      nsets = 10,
      nintersects = 15,
      keep.order = TRUE,
      sets=c("pool_G", "pool_20"),
      mainbar.y.label = "Number of shared proteins",
      sets.x.label = "Number of proteins per fraction",
      set_size.show = TRUE,
      set_size.angles = 0,
      set_size.scale_max = 1000
)

#venn digarams
#get colors
D1_col <- rgb(255,93,93, maxColorValue = 255)
D2_col <- rgb(169,208,142,maxColorValue = 255)
#E4_D1_pool_col <- rgb(255,217,102, maxColorValue = 255)

ggvenn(lst,
       fill_color = c(D1_col,D2_col),
       stroke_size = 0.5, set_name_size = 4, text_size = 4, digits = 0, fill_alpha = 0.9)
```

### show fasta headers of non-shared proteins

```{r}
#E4_D1_Pool_G_B1-B9
data <- readRDS(file = "results_v002/results_run1_mqpar_extracts_2gether_E4_D1_B_LFQ_combined_txt_proteinGroups.txt_Unique.RDS")
data <- data[[1]]
data_sub <- dplyr::select(data, contains("Pool_G"))
B_D1<-apply(data_sub, 1, any)
fasta <- data$FASTA
B_D1 <- as.data.frame(cbind(B_D1, fasta))
#B_D1 <- as.data.frame(cbind(B_D1, data$FASTA))

#E4_D1_Pool_20_B1-B9
D1_raw <- readRDS(file = "results_v002/results_run1_mqpar_extracts_2gether_E4_D1_B_LFQ_combined_txt_proteinGroups.txt_Unique.RDS")
D1_raw <- D1_raw[[1]]
fasta <- D1_raw$FASTA
D1 <- dplyr::select(D1_raw, contains("20"))
D1 <- apply(D1, 1, any)
D1 <- as.data.frame(cbind(D1, fasta))

G_pool_comparison  <- merge(B_D1, D1,
                            all = TRUE,
                            by = "fasta")
tmp <- G_pool_comparison
G_pool_comparison$B_D1 <- as.logical(G_pool_comparison$B_D1)
G_pool_comparison$D1 <- as.logical(G_pool_comparison$D1)
G_pool_comparison <- G_pool_comparison[,-1]*1
G_pool_comparison[G_pool_comparison==0] <- NA

proteins <- as.data.frame(which(!is.na(G_pool_comparison), arr.ind=TRUE))

B_D1_inter <- proteins %>% dplyr::filter(col == 1)
D1_inter <- proteins %>% dplyr::filter(col == 2)

# combine to list
lst <- list("pool_G"=as.double(B_D1_inter[,1]),
            "pool_20"=as.double(D1_inter[,1]))

test <- as.data.frame(tmp$fasta[setdiff(lst[[1]],lst[[2]])])
test
```


# Compare E3e E3f with E4_D1_a-j

-   E3 e bzw E3 f sollte dasselbe sein wie das hier
    `file:///X:/wissenschaftliche%20Ver%C3%B6ffentlichungen/Publikationen/2022/Proteomics_JRS_DL/data/results_v002/results_run1_mqpar_extracts_2gether_E4_D1_a-j_LFQ_combined_txt_proteinGroups.txt_post-processing.html#52_Summary_Unique`
    und auch E5_D1 a und b

```{r}
aj_raw <- readRDS(file = "results_v002/results_run1_mqpar_extracts_2gether_E4_D1_a-j_LFQ_combined_txt_proteinGroups.txt_Unique.RDS")
aj_raw <- aj_raw[[1]]
aj <- aj_raw[,-ncol(aj_raw)]
FASTA <- c()
for (i in 1:length(aj_raw$FASTA)){
  tmp <- strsplit(aj_raw$FASTA[i],split='\\|')[[1]][2]
  FASTA<- c(FASTA,tmp)
}
aj <- cbind(aj,FASTA)

E3_D1_raw <- readRDS(file = "results_v002/results_run1_mqpar_extracts_2gether_E3_D1_LFQ_combined_txt_proteinGroups.txt_Unique.RDS")
E3_D1_raw <- E3_D1_raw[[1]]
E3_D1 <- E3_D1_raw[,c(13:18)]
FASTA <- c()
for (i in 1:length(E3_D1_raw$FASTA)){
  tmp <- strsplit(E3_D1_raw$FASTA[i],split='\\|')[[1]][2]
  FASTA<- c(FASTA,tmp)
}
E3_D1 <- cbind(E3_D1,FASTA)

E5_D1_raw <- readRDS(file="./27_20230201_FH/results/D5_extracts_Unique.RDS")
E5_D1_raw <- E5_D1_raw[[1]]
E5_D1 <- dplyr::select(E5_D1_raw, contains(".1"))
colnames(E5_D1) <- c(paste("Unique.peptides.","E5_D",substring(colnames(E5_D1),17),sep=""))
FASTA <- c()
for (i in 1:length(E5_D1_raw$FASTA)){
  tmp <- strsplit(E5_D1_raw$FASTA[i],split='\\|')[[1]][2]
  FASTA<- c(FASTA,tmp)
}
E5_D1 <- cbind(E5_D1,FASTA)


comp  <- merge(aj, E3_D1,
               all = TRUE,
               by = "FASTA")
comp <- merge(comp, E5_D1,
              all=TRUE,
              by="FASTA")


paletteLength <- 2
myColor <- colorRampPalette(c("navy", "white",
                              "firebrick3"))(paletteLength)
data4pheatmap <- comp[,-1]
data4pheatmap <- data4pheatmap*1
data4pheatmap[data4pheatmap == 0] <- NA
# remove NA rows
ind <- apply(data4pheatmap, 1, function(x) all(is.na(x)))
data4pheatmap_clear <- data4pheatmap[!ind,]
colnames(data4pheatmap_clear) <- colnames(comp[-1])
data4pheatmap_clear[is.na(data4pheatmap_clear)] <- 0
pheatmap(data4pheatmap_clear,
         legend_breaks = c(0,1),
         color = myColor,
         treeheight_row = 10,
         angle_col ="45",
         treeheight_col = 10,
         legend = TRUE,
         labels_row = rep("",nrow(data4pheatmap)),
         labels_col = substring(colnames(data4pheatmap_clear),17))
#plot results
results_2gether <- sapply(comp[,-1],function(x)sum(x, na.rm = TRUE))
print(kable(results_2gether))
ggplot(melt(results_2gether),aes(x = rownames(melt(results_2gether)), y = value)) +
  geom_boxplot() + 
  xlab("Sample name") + 
  scale_x_discrete(labels = substring(colnames(data4pheatmap_clear),17)) +
  theme(axis.text.x = element_text(angle = 90)) +
  labs(title = "Unique")
```

# D2

```{r}
#############################
# D2
###############################
data <- readRDS(file = "results_v002/results_run1_mqpar_extracts_2gether_D2_LFQ_combined_txt_proteinGroups.txt_Unique.RDS")
data <- data[[1]]
data_sub <- dplyr::select(data, contains("Pool_20"))
B_D2 <- data_sub*1
B_D2[B_D2==0] <- NA


### analysis B
proteins <- as.data.frame(which(!is.na(B_D2), arr.ind=TRUE))

B1 <- proteins %>% dplyr::filter(col == 1 ) # | col == 2 -> means or col2
B10 <- proteins %>% dplyr::filter(col == 2)
B2 <- proteins %>% dplyr::filter(col == 3)
B3 <- proteins %>% dplyr::filter(col == 4)
B4 <- proteins %>% dplyr::filter(col == 5)
B5 <- proteins %>% dplyr::filter(col == 6)
B6 <- proteins %>% dplyr::filter(col == 7)
B7 <- proteins %>% dplyr::filter(col == 8)
B8 <- proteins %>% dplyr::filter(col == 9)
B9 <- proteins %>% dplyr::filter(col == 10)
# combine to list
lst <- list(B1 = as.double(B1[,1]),
            B2 = as.double(B2[,1]),
            B3 = as.double(B3[,1]),
            B4 = as.double(B4[,1]),
            B5 = as.double(B5[,1]),
            B6 = as.double(B6[,1]),
            B7 = as.double(B7[,1]),
            B8 = as.double(B8[,1]),
            B9 = as.double(B9[,1]),
            B10 = as.double(B10[,1]))            

upset(fromList(lst), 
      order.by = "freq",
      nsets = 9,
      nintersects = 15,
      sets = c("B1","B2","B3","B4","B5","B6","B7","B8","B9","B10"),
      keep.order = TRUE,
      mainbar.y.label = "Number of shared proteins",
      sets.x.label = "Number of proteins per fraction",
      set_size.show = TRUE,
      set_size.angles = 0,
      set_size.scale_max = 1000
)


data_D2_pool <- dplyr::select(data, contains("Gesamt"))
data_D2_pool <- apply(data_D2_pool, 1, any)
data_D2_fractions <- dplyr::select(data, contains("20"))
data_D2_fractions <- apply(data_D2_fractions, 1, any)
data_D2 <- as.data.frame(cbind(data_D2_pool,data_D2_fractions))
data_D2 <- data_D2*1
data_D2[data_D2==0] <- NA
# boxplot
boxplot(sum(data_D2$data_D2_pool,na.rm = TRUE),
        sum(data_D2$data_D2_fractions,na.rm = TRUE),
        names = c("D2 pool","D2 fractions"))


proteins <- as.data.frame(which(!is.na(data_D2), arr.ind=TRUE))

D2_pool_inter <- proteins %>% dplyr::filter(col == 1)
D2_fractions_inter <- proteins %>% dplyr::filter(col == 2)

# combine to list
lst <- list(D2_pool=as.double(D2_pool_inter[,1]),
            D2_fractions=as.double(D2_fractions_inter[,1]))

upset(fromList(lst), 
      order.by = "freq",
      nsets = 10,
      nintersects = 15,
      keep.order = TRUE,
      sets=c("D2_pool", "D2_fractions"),
      mainbar.y.label = "Number of shared proteins",
      sets.x.label = "Number of proteins per fraction",
      set_size.show = TRUE,
      set_size.angles = 0,
      set_size.scale_max = 1000
)

#venn digarams
#get colors
D2_pool_col <- rgb(255,93,93, maxColorValue = 255)
D2_ingel_col <- rgb(169,208,142,maxColorValue = 255)
#E4_D1_pool_col <- rgb(255,217,102, maxColorValue = 255)

ggvenn(lst,
       fill_color = c(D2_pool_col,D2_ingel_col),
       stroke_size = 0.5, set_name_size = 5, text_size = 4, digits = 0, fill_alpha = 0.9)


```

# Compare D1-D2

## fractions

### E4_D1_Pool_G\_B1-B9 with E4_D2_Pool_20_B1-B9

```{r}
#################### D1 fractions
data <- readRDS(file = "results_v002/results_run1_mqpar_extracts_2gether_E4_D1_B_LFQ_combined_txt_proteinGroups.txt_Unique.RDS")
data <- data[[1]]
data_sub <- dplyr::select(data, contains("Pool_G"))
B_D1<-apply(data_sub, 1, any)
fasta <- c()
for (i in 1:length(data$FASTA)){
  tmp <- strsplit(data$FASTA[i],split='\\|')[[1]][2]
  fasta <- c(fasta,tmp)
}
B_D1 <- as.data.frame(cbind(B_D1, fasta))
#B_D1 <- as.data.frame(cbind(B_D1, data$FASTA))


#################### D2 fractions
data <- readRDS(file = "results_v002/results_run1_mqpar_extracts_2gether_D2_LFQ_combined_txt_proteinGroups.txt_Unique.RDS")
data <- data[[1]]
data_sub <- dplyr::select(data, contains("Pool_20"))
B_D2<-apply(data_sub, 1, any)
fasta <- c()
for (i in 1:length(data$FASTA)){
  tmp <- strsplit(data$FASTA[i],split='\\|')[[1]][2]
  fasta <- c(fasta,tmp)
}
B_D2 <- as.data.frame(cbind(B_D2, fasta))
#B_D2 <- as.data.frame(cbind(B_D2, data$FASTA))
#D1_D2_comparison  <- merge(B_D1, B_D2,
#                           all = TRUE,
#                           by = "V2")
D1_D2_comparison  <- merge(B_D1, B_D2,
                           all = TRUE,
                           by = "fasta")

D1_D2_comparison$B_D1 <- as.logical(D1_D2_comparison$B_D1)
D1_D2_comparison$B_D2 <- as.logical(D1_D2_comparison$B_D2)
D1_D2_comparison <- D1_D2_comparison[,-1]*1
D1_D2_comparison[D1_D2_comparison==0] <- NA

# boxplot
boxplot(sum(D1_D2_comparison$B_D1,na.rm = TRUE),
        sum(D1_D2_comparison$B_D2,na.rm = TRUE),
        names = c("D1","D2"),
        main="Comparison D1/D2 fractions")
#upset r & ggvenn
proteins <- as.data.frame(which(!is.na(D1_D2_comparison), arr.ind=TRUE))

D1_inter <- proteins %>% dplyr::filter(col == 1)
D2_inter <- proteins %>% dplyr::filter(col == 2)

# combine to list
lst <- list("D1\ningeldigest_fraction"=as.double(D1_inter[,1]),
            "D2\ningeldigest_fraction"=as.double(D2_inter[,1]))

upset(fromList(lst), 
      order.by = "freq",
      nsets = 10,
      nintersects = 15,
      keep.order = TRUE,
      sets=c("D1\ningeldigest_fraction", "D2\ningeldigest_fraction"),
      mainbar.y.label = "Number of shared proteins",
      sets.x.label = "Number of proteins per fraction",
      set_size.show = TRUE,
      set_size.angles = 0,
      set_size.scale_max = 1000
)

#venn digarams
#get colors
D1_col <- rgb(255,93,93, maxColorValue = 255)
D2_col <- rgb(169,208,142,maxColorValue = 255)
#E4_D1_pool_col <- rgb(255,217,102, maxColorValue = 255)

ggvenn(lst,
       fill_color = c(D1_col,D2_col),
       stroke_size = 0.5, set_name_size = 4, text_size = 4, digits = 0, fill_alpha = 0.9)
```

### E4_D1_Pool_20_B1-B9 with E4_D2_Pool_20_B1-B9

```{r}
# D1-D2 comparison

D1_raw <- readRDS(file = "results_v002/results_run1_mqpar_extracts_2gether_E4_D1_B_LFQ_combined_txt_proteinGroups.txt_Unique.RDS")
D1_raw <- D1_raw[[1]]
D1 <- dplyr::select(D1_raw, contains("20"))
D1 <- apply(D1, 1, any)
#D1 <- as.data.frame(cbind(D1, D1_raw$FASTA))
fasta <- c()
for (i in 1:length(data$FASTA)){
  tmp <- strsplit(data$FASTA[i],split='\\|')[[1]][2]
  fasta <- c(fasta,tmp)
}
D1 <- as.data.frame(cbind(D1, fasta))
D2_raw <- readRDS(file = "results_v002/results_run1_mqpar_extracts_2gether_D2_LFQ_combined_txt_proteinGroups.txt_Unique.RDS")
D2_raw <- D2_raw[[1]]
D2 <- dplyr::select(D2_raw, contains("Pool_20"))
D2 <- apply(D2, 1, any)
fasta <- c()
for (i in 1:length(data$FASTA)){
  tmp <- strsplit(data$FASTA[i],split='\\|')[[1]][2]
  fasta <- c(fasta,tmp)
}
#D2 <- as.data.frame(cbind(D2, D2_raw$FASTA))
D2 <- as.data.frame(cbind(D2, fasta))
D1_D2_comparison  <- merge(D1, D2,
                           all = TRUE,
                           by = "fasta")

D1_D2_comparison$D1 <- as.logical(D1_D2_comparison$D1)
D1_D2_comparison$D2 <- as.logical(D1_D2_comparison$D2)
D1_D2_comparison <- D1_D2_comparison[,-1]*1
D1_D2_comparison[D1_D2_comparison==0] <- NA

# boxplot
boxplot(sum(D1_D2_comparison$D1,na.rm = TRUE),
        sum(D1_D2_comparison$D2,na.rm = TRUE),
        names = c("D1","D2"))

proteins <- as.data.frame(which(!is.na(D1_D2_comparison), arr.ind=TRUE))

D1_inter <- proteins %>% dplyr::filter(col == 1)
D2_inter <- proteins %>% dplyr::filter(col == 2)

# combine to list
lst <- list("D1\ningeldigest_pool"=as.double(D1_inter[,1]),
            "D2\ningeldigest_pool"=as.double(D2_inter[,1]))

upset(fromList(lst), 
      order.by = "freq",
      nsets = 10,
      nintersects = 15,
      keep.order = TRUE,
      sets=c("D1\ningeldigest_pool", "D2\ningeldigest_pool"),
      mainbar.y.label = "Number of shared proteins",
      sets.x.label = "Number of proteins per fraction",
      set_size.show = TRUE,
      set_size.angles = 0,
      set_size.scale_max = 1000
)

#venn digarams
#get colors
D1_col <- rgb(255,93,93, maxColorValue = 255)
D2_col <- rgb(169,208,142,maxColorValue = 255)
#E4_D1_pool_col <- rgb(255,217,102, maxColorValue = 255)

ggvenn(lst,
       fill_color = c(D1_col,D2_col),
       stroke_size = 0.5, set_name_size = 4, text_size = 4, digits = 0, fill_alpha = 0.9)
```

## pool

```{r}
# D1-D2 comparison

D1_raw <- readRDS(file = "results_v002/results_run1_mqpar_extracts_2gether_E4_D1_a-j_LFQ_combined_txt_proteinGroups.txt_Unique.RDS")
D1_raw <- D1_raw[[1]]
D1 <- dplyr::select(D1_raw, contains("Pool"))
D1 <- apply(D1, 1, any)
#D1 <- as.data.frame(cbind(D1, D1_raw$FASTA))
fasta <- c()
for (i in 1:length(D1_raw$FASTA)){
  tmp <- strsplit(D1_raw$FASTA[i],split='\\|')[[1]][2]
  fasta <- c(fasta,tmp)
}
D1 <- as.data.frame(cbind(D1, fasta))
D2_raw <- readRDS(file = "results_v002/results_run1_mqpar_extracts_2gether_D2_LFQ_combined_txt_proteinGroups.txt_Unique.RDS")
D2_raw <- D2_raw[[1]]
D2 <- dplyr::select(D2_raw, contains("Gesamt"))
D2 <- apply(D2, 1, any)
fasta <- c()
for (i in 1:length(D2_raw$FASTA)){
  tmp <- strsplit(D2_raw$FASTA[i],split='\\|')[[1]][2]
  fasta <- c(fasta,tmp)
}
#D2 <- as.data.frame(cbind(D2, D2_raw$FASTA))
D2 <- as.data.frame(cbind(D2, fasta))
D1_D2_comparison  <- merge(D1, D2,
                           all = TRUE,
                           by = "fasta")

D1_D2_comparison$D1 <- as.logical(D1_D2_comparison$D1)
D1_D2_comparison$D2 <- as.logical(D1_D2_comparison$D2)
D1_D2_comparison <- D1_D2_comparison[,-1]*1
D1_D2_comparison[D1_D2_comparison==0] <- NA

# boxplot
boxplot(sum(D1_D2_comparison$D1,na.rm = TRUE),
        sum(D1_D2_comparison$D2,na.rm = TRUE),
        names = c("D1","D2"))

proteins <- as.data.frame(which(!is.na(D1_D2_comparison), arr.ind=TRUE))

D1_inter <- proteins %>% dplyr::filter(col == 1)
D2_inter <- proteins %>% dplyr::filter(col == 2)

# combine to list
lst <- list("D1\ningeldigest_pool"=as.double(D1_inter[,1]),
            "D2\ningeldigest_pool"=as.double(D2_inter[,1]))

upset(fromList(lst), 
      order.by = "freq",
      nsets = 10,
      nintersects = 15,
      keep.order = TRUE,
      sets=c("D1\ningeldigest_pool", "D2\ningeldigest_pool"),
      mainbar.y.label = "Number of shared proteins",
      sets.x.label = "Number of proteins per fraction",
      set_size.show = TRUE,
      set_size.angles = 0,
      set_size.scale_max = 1000
)

#venn digarams
#get colors
D1_col <- rgb(255,93,93, maxColorValue = 255)
D2_col <- rgb(169,208,142,maxColorValue = 255)
#E4_D1_pool_col <- rgb(255,217,102, maxColorValue = 255)

ggvenn(lst,
       fill_color = c(D1_col,D2_col),
       stroke_size = 0.5, set_name_size = 4, text_size = 4, digits = 0, fill_alpha = 0.9)
```

# E5 - venn diagramms of different dimensions

-   `file:///N:/1_A_Bachelor_Master_Intern/2_M_2022/David/Data/27_20230201_FH/post-processing-extracts.html#52_Summary_Unique`
-   venn diagramm 1a,2a,3a,
-   venn diagramm 1b,2b,3b,4b

## a

```{r}
data <- readRDS("./27_20230201_FH/results/D5_extracts_Razor.RDS")
data <- data[[1]]
a <- dplyr::select(data, contains("_a"))
D1_a <- a$Razor...unique.peptides.1_a_01+a$Razor...unique.peptides.1_a_02
D2_a <- a$Razor...unique.peptides.2_a_01 + a$Razor...unique.peptides.2_a_02
D3_a <- a$Razor...unique.peptides.3_a_01 + a$Razor...unique.peptides.3_a_02
a <- as.data.frame(cbind(D1_a,D2_a,D3_a))
a[a==0] <- NA
a[a==2] <- 1
boxplot(sum(a$Razor...unique.peptides.1_a_01,na.rm = TRUE),
        sum(a$Razor...unique.peptides.2_a_01,na.rm = TRUE),
        sum(a$Razor...unique.peptides.3_a_01,na.rm = TRUE),
        names = c("D1","D2","D3"),
        main="unique peptides E5_a D1-D3")
#venn diagram
proteins <- as.data.frame(which(!is.na(a), arr.ind=TRUE))
D1_a <- proteins %>% dplyr::filter(col == 1)
D2_a <- proteins %>% dplyr::filter(col == 2)
D3_a <- proteins %>% dplyr::filter(col == 3)

# combine to list
lst <- list(D1_a = as.double(D1_a[,1]),
            D2_a = as.double(D2_a[,1]),
            D3_a = as.double(D3_a[,1]))
#upset r
upset(fromList(lst), 
      order.by = "freq",
      nsets = 9,
      nintersects = 15,
      sets = c("D1_a","D2_a","D3_a"),
      keep.order = TRUE,
      mainbar.y.label = "Number of shared proteins",
      sets.x.label = "Number of proteins per fraction",
      set_size.show = TRUE,
      set_size.angles = 0,
      set_size.scale_max = 1500
)
# venn diagram
ggvenn(lst,
       stroke_size = 0.5, set_name_size = 5, text_size = 4.25, digits = 0, fill_alpha = 0.9, label_sep = )
```

## b

```{r}
data <- readRDS("./27_20230201_FH/results/D5_extracts_Razor.RDS")
data <- data[[1]]
b <- dplyr::select(data, contains("_b"))
D1_b <- b$Razor...unique.peptides.1_b_01+ b$Razor...unique.peptides.1_b_02
D2_b <- b$Razor...unique.peptides.2_b_01 + b$Razor...unique.peptides.2_b_02
D3_b <- b$Razor...unique.peptides.3_b_01 + b$Razor...unique.peptides.3_b_02
D4_b <- b$Razor...unique.peptides.3_b_01 + b$Razor...unique.peptides.4_b_02
b <- as.data.frame(cbind(D1_b,D2_b,D3_b,D4_b))
b[b==0] <- NA
b[b==2] <- 1
boxplot(sum(b$Razor...unique.peptides.1_b_01,na.rm = TRUE),
        sum(b$Razor...unique.peptides.2_b_01,na.rm = TRUE),
        sum(b$Razor...unique.peptides.3_b_01,na.rm = TRUE),
        sum(b$Razor...unique.peptides.4_b_01,na.rm = TRUE),
        names = c("D1","D2","D3","D4"),
        main="unique peptides E5_b D1-D4")
#venn diagram
proteins <- as.data.frame(which(!is.na(b), arr.ind=TRUE))
D1_b <- proteins %>% dplyr::filter(col == 1)
D2_b <- proteins %>% dplyr::filter(col == 2)
D3_b <- proteins %>% dplyr::filter(col == 3)
D4_b <- proteins %>% dplyr::filter(col == 4)

# combine to list
lst <- list(D1_b = as.double(D1_b[,1]),
            D2_b = as.double(D2_b[,1]),
            D3_b = as.double(D3_b[,1]),
            D4_b = as.double(D4_b[,1]))
#upset r
upset(fromList(lst), 
      order.by = "freq",
      nsets = 9,
      nintersects = 15,
      sets = c("D1_b","D2_b","D3_b", "D4_b"),
      keep.order = TRUE,
      mainbar.y.label = "Number of shared proteins",
      sets.x.label = "Number of proteins per fraction",
      set_size.show = TRUE,
      set_size.angles = 0,
      set_size.scale_max = 1500
)
# venn diagram
ggvenn(lst,
       stroke_size = 0.5, set_name_size = 5, text_size = 4.25, digits = 0, fill_alpha = 0.9, label_sep = )
```
